pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4082689&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4082689&tab=artifacts#/) on master @ [a23cf94c9bf72a77cd08037df082a2a3f2e5d948](https://github.com/cockroachdb/cockroach/commits/a23cf94c9bf72a77cd08037df082a2a3f2e5d948):


```
=== RUN   TestRefresh
--- FAIL: TestRefresh (0.31s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:141: 
        	Error Trace:	cache_test.go:471
        	            				cache_test.go:141
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-12171

kv/kvserver/protectedts/ptcache.TestRefresh [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4092098&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4092098&tab=artifacts#/) on master @ [8c06168d3183a6b983d97a468cdb915d264086f8](https://github.com/cockroachdb/cockroach/commits/8c06168d3183a6b983d97a468cdb915d264086f8):


```
I220111 13:41:50.738912 1333 server/init.go:264  [n?] 90  active cluster version: 21.2-36
W220111 13:41:50.739033 1333 1@gossip/gossip.go:1472  [n1] 91  no addresses found; use --join to specify a connected node
I220111 13:41:50.739123 1427 1@server/server.go:1620  [n1] 92  connecting to gossip network to verify cluster ID "4e72549f-e057-40d0-922f-ff4b9a896669"
I220111 13:41:50.739122 1333 gossip/gossip.go:401  [n1] 93  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34647" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:2 patch:0 internal:36 > build_tag:"v22.1.0-alpha.00000000-2776-g8c06168d31" started_at:1641908510739115419 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:45653" >
I220111 13:41:50.742835 1333 server/node.go:451  [n1] 94  initialized store s1
I220111 13:41:50.742898 1333 kv/kvserver/stores.go:250  [n1] 95  read 0 node addresses from persistent storage
I220111 13:41:50.743051 1333 server/node.go:528  [n1] 96  started with engine type 2
I220111 13:41:50.743113 1333 server/node.go:530  [n1] 97  started with attributes []
I220111 13:41:50.743221 1333 1@server/server.go:1764  [n1] 98  starting https server at 127.0.0.1:40515 (use: 127.0.0.1:40515)
I220111 13:41:50.743244 1333 1@server/server.go:1769  [n1] 99  starting postgres server at 127.0.0.1:45653 (use: 127.0.0.1:45653)
I220111 13:41:50.743268 1333 1@server/server.go:1771  [n1] 100  starting grpc server at 127.0.0.1:34647
I220111 13:41:50.743292 1333 1@server/server.go:1772  [n1] 101  advertising CockroachDB node at 127.0.0.1:34647
I220111 13:41:50.744596 1427 1@server/server.go:1623  [n1] 102  node connected via gossip
W220111 13:41:50.779795 1294 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 103  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W220111 13:41:50.782510 1293 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 104  range lookup of key /Table/20 found only non-matching ranges []; retrying
I220111 13:41:50.809393 1333 1@util/log/event_log.go:32  [n1] 105 ={"Timestamp":1641908510809390367,"EventType":"node_join","NodeID":1,"StartedAt":1641908510739115419,"LastUp":1641908510739115419}
I220111 13:41:50.809466 1333 sql/sqlliveness/slinstance/slinstance.go:300  [n1] 106  starting SQL liveness instance
I220111 13:41:50.810660 1546 sql/sqlliveness/slstorage/slstorage.go:406  [n1] 107  inserted sqlliveness session 6aaf25ffea454c8883e6c731e7828f98
I220111 13:41:50.810722 1546 sql/sqlliveness/slinstance/slinstance.go:187  [n1] 108  created new SQL liveness session 6aaf25ffea454c8883e6c731e7828f98
I220111 13:41:50.811100 1439 sql/temporary_schema.go:515  [n1] 109  running temporary object cleanup background job
I220111 13:41:50.811619 1439 sql/temporary_schema.go:559  [n1] 110  found 0 temporary schemas
I220111 13:41:50.811640 1439 sql/temporary_schema.go:562  [n1] 111  early exiting temporary schema cleaner as no temporary schemas were found
I220111 13:41:50.811657 1439 sql/temporary_schema.go:563  [n1] 112  completed temporary object cleanup job
I220111 13:41:50.811670 1439 sql/temporary_schema.go:646  [n1] 113  temporary object cleaner next scheduled to run at 2022-01-11 14:11:50.811062505 +0000 UTC
I220111 13:41:50.811645 1441 sql/sqlstats/persistedsqlstats/provider.go:137  [n1] 114  starting sql-stats-worker with initial delay: 55m21.958734138s
I220111 13:41:50.819849 1642 kv/kvserver/replica_rangefeed.go:651  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 115  RangeFeed closed timestamp is empty
I220111 13:41:50.827947 1749 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 116 ={"Timestamp":1641908510820652253,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I220111 13:41:50.831228 1849 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 117  no need to migrate, cluster already at newest version
I220111 13:41:50.832573 1849 util/log/event_log.go:32  [n1,intExec=set-setting] 118 ={"Timestamp":1641908510830070232,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.2-36'"],"SettingName":"version","Value":"21.2-36"}
I220111 13:41:50.847661 2147 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 119 ={"Timestamp":1641908510844451105,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"2d6ce742-d9c5-47f5-9567-f9fd3367e04a"}
I220111 13:41:50.852535 2137 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 120 ={"Timestamp":1641908510849118666,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I220111 13:41:50.855469 1449 3@vendor/github.com/cockroachdb/pebble/event.go:677  [n1,pebble,s1] 121  [JOB 3] WAL created 000004
I220111 13:41:50.856387 2252 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 122 ={"Timestamp":1641908510852750170,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I220111 13:41:50.881695 1333 server/server_sql.go:1161  [n1] 123  done ensuring all necessary startup migrations have run
I220111 13:41:50.881901 2158 jobs/job_scheduler.go:357  [n1] 124  waiting 2m0s before scheduled jobs daemon start
I220111 13:41:50.883947 2160 server/auto_upgrade.go:55  [n1] 125  no need to upgrade, cluster already at the newest version
I220111 13:41:50.895220 1333 1@server/server.go:2448  [n1] 126  serving sql connections
I220111 13:41:50.895818 1333 testutils/testcluster/testcluster.go:1158  [-] 127  WaitForFullReplication
I220111 13:41:50.895856 1333 testutils/testcluster/testcluster.go:1162  [-] 128  WaitForFullReplication took: 225ns
I220111 13:41:50.895904 1333 2@rpc/context.go:1175  [n1] 129  dialing n1: 127.0.0.1:34647 (default)
--- FAIL: TestRefresh (0.32s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:141: 
        	Error Trace:	cache_test.go:471
        	            				cache_test.go:141
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptcache.TestRefresh [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4127614&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4127614&tab=artifacts#/) on master @ [78419450178335b31f542bd1b14fefdf4ecee0e8](https://github.com/cockroachdb/cockroach/commits/78419450178335b31f542bd1b14fefdf4ecee0e8):


```
I220115 08:19:08.404147 1305 server/init.go:264  [n?] 90  active cluster version: 21.2-36
W220115 08:19:08.404216 1305 1@gossip/gossip.go:1472  [n1] 91  no addresses found; use --join to specify a connected node
I220115 08:19:08.404286 1376 1@server/server.go:1620  [n1] 92  connecting to gossip network to verify cluster ID "dbe40fb9-7e9e-41ff-874d-6f745ee85497"
I220115 08:19:08.404267 1305 gossip/gossip.go:401  [n1] 93  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45223" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:2 patch:0 internal:36 > build_tag:"v22.1.0-alpha.00000000-2941-g7841945017" started_at:1642234748404262592 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:45667" >
I220115 08:19:08.406882 1305 server/node.go:451  [n1] 94  initialized store s1
I220115 08:19:08.406930 1305 kv/kvserver/stores.go:250  [n1] 95  read 0 node addresses from persistent storage
I220115 08:19:08.407001 1305 server/node.go:528  [n1] 96  started with engine type 2
I220115 08:19:08.407022 1305 server/node.go:530  [n1] 97  started with attributes []
I220115 08:19:08.407092 1305 1@server/server.go:1764  [n1] 98  starting https server at 127.0.0.1:35217 (use: 127.0.0.1:35217)
I220115 08:19:08.407119 1305 1@server/server.go:1769  [n1] 99  starting postgres server at 127.0.0.1:45667 (use: 127.0.0.1:45667)
I220115 08:19:08.407135 1305 1@server/server.go:1771  [n1] 100  starting grpc server at 127.0.0.1:45223
I220115 08:19:08.407161 1305 1@server/server.go:1772  [n1] 101  advertising CockroachDB node at 127.0.0.1:45223
I220115 08:19:08.412021 1376 1@server/server.go:1623  [n1] 102  node connected via gossip
W220115 08:19:08.445293 1264 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 103  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W220115 08:19:08.447829 1263 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 104  range lookup of key /Table/20 found only non-matching ranges []; retrying
I220115 08:19:08.471491 1305 1@util/log/event_log.go:32  [n1] 105 ={"Timestamp":1642234748471488092,"EventType":"node_join","NodeID":1,"StartedAt":1642234748404262592,"LastUp":1642234748404262592}
I220115 08:19:08.471562 1305 sql/sqlliveness/slinstance/slinstance.go:300  [n1] 106  starting SQL liveness instance
I220115 08:19:08.473299 1388 sql/temporary_schema.go:515  [n1] 107  running temporary object cleanup background job
I220115 08:19:08.473381 1539 sql/sqlliveness/slstorage/slstorage.go:406  [n1] 108  inserted sqlliveness session 715450b0c3e64c7dbf11ef2abb96bd08
I220115 08:19:08.473421 1539 sql/sqlliveness/slinstance/slinstance.go:187  [n1] 109  created new SQL liveness session 715450b0c3e64c7dbf11ef2abb96bd08
I220115 08:19:08.473422 1390 sql/sqlstats/persistedsqlstats/provider.go:137  [n1] 110  starting sql-stats-worker with initial delay: 53m13.648399661s
I220115 08:19:08.473877 1388 sql/temporary_schema.go:559  [n1] 111  found 0 temporary schemas
I220115 08:19:08.473909 1388 sql/temporary_schema.go:562  [n1] 112  early exiting temporary schema cleaner as no temporary schemas were found
I220115 08:19:08.473922 1388 sql/temporary_schema.go:563  [n1] 113  completed temporary object cleanup job
I220115 08:19:08.473934 1388 sql/temporary_schema.go:646  [n1] 114  temporary object cleaner next scheduled to run at 2022-01-15 08:49:08.473280312 +0000 UTC
I220115 08:19:08.482626 1700 kv/kvserver/replica_rangefeed.go:651  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 115  RangeFeed closed timestamp is empty
I220115 08:19:08.487647 1714 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 116 ={"Timestamp":1642234748481910233,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I220115 08:19:08.490832 1710 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 117  no need to migrate, cluster already at newest version
I220115 08:19:08.492254 1710 util/log/event_log.go:32  [n1,intExec=set-setting] 118 ={"Timestamp":1642234748489779087,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.2-36'"],"SettingName":"version","Value":"21.2-36"}
I220115 08:19:08.506569 2016 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 119 ={"Timestamp":1642234748501583182,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"51e40f77-f58e-47ff-8a69-df991f47270f"}
I220115 08:19:08.511375 2101 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 120 ={"Timestamp":1642234748508598096,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I220115 08:19:08.515566 1887 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 121 ={"Timestamp":1642234748511571199,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I220115 08:19:08.520750 1431 3@vendor/github.com/cockroachdb/pebble/event.go:677  [n1,pebble,s1] 122  [JOB 3] WAL created 000004
I220115 08:19:08.538815 1305 server/server_sql.go:1166  [n1] 123  done ensuring all necessary startup migrations have run
I220115 08:19:08.538903 2438 jobs/job_scheduler.go:357  [n1] 124  waiting 2m0s before scheduled jobs daemon start
I220115 08:19:08.540771 2440 server/auto_upgrade.go:55  [n1] 125  no need to upgrade, cluster already at the newest version
I220115 08:19:08.549715 1305 1@server/server.go:2448  [n1] 126  serving sql connections
I220115 08:19:08.550227 1305 testutils/testcluster/testcluster.go:1158  [-] 127  WaitForFullReplication
I220115 08:19:08.550256 1305 testutils/testcluster/testcluster.go:1162  [-] 128  WaitForFullReplication took: 295ns
I220115 08:19:08.550306 1305 2@rpc/context.go:1179  [n1] 129  dialing n1: 127.0.0.1:45223 (default)
--- FAIL: TestRefresh (0.29s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:141: 
        	Error Trace:	cache_test.go:471
        	            				cache_test.go:141
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptcache.TestRefresh [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4132437&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4132437&tab=artifacts#/) on master @ [5ad21e3896ee809e9c3ebc28bb22166f1275acca](https://github.com/cockroachdb/cockroach/commits/5ad21e3896ee809e9c3ebc28bb22166f1275acca):


```
I220116 08:29:59.926258 1314 server/init.go:264  [n?] 89  active cluster version: 21.2-36
W220116 08:29:59.926325 1314 1@gossip/gossip.go:1472  [n1] 90  no addresses found; use --join to specify a connected node
I220116 08:29:59.926382 1393 1@server/server.go:1620  [n1] 91  connecting to gossip network to verify cluster ID "1b466a47-377a-4944-b994-ea100cf521a5"
I220116 08:29:59.926367 1314 gossip/gossip.go:401  [n1] 92  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37681" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:2 patch:0 internal:36 > build_tag:"v22.1.0-alpha.00000000-2945-g5ad21e3896" started_at:1642321799926364070 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:45091" >
I220116 08:29:59.929698 1314 server/node.go:451  [n1] 93  initialized store s1
I220116 08:29:59.929780 1314 kv/kvserver/stores.go:250  [n1] 94  read 0 node addresses from persistent storage
I220116 08:29:59.929855 1314 server/node.go:528  [n1] 95  started with engine type 2
I220116 08:29:59.929893 1314 server/node.go:530  [n1] 96  started with attributes []
I220116 08:29:59.930033 1314 1@server/server.go:1764  [n1] 97  starting https server at 127.0.0.1:33585 (use: 127.0.0.1:33585)
I220116 08:29:59.930090 1314 1@server/server.go:1769  [n1] 98  starting postgres server at 127.0.0.1:45091 (use: 127.0.0.1:45091)
I220116 08:29:59.930130 1314 1@server/server.go:1771  [n1] 99  starting grpc server at 127.0.0.1:37681
I220116 08:29:59.930170 1314 1@server/server.go:1772  [n1] 100  advertising CockroachDB node at 127.0.0.1:37681
I220116 08:29:59.930300 1393 1@server/server.go:1623  [n1] 101  node connected via gossip
W220116 08:29:59.964790 1002 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 102  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W220116 08:29:59.966069 1001 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 103  range lookup of key /Table/20 found only non-matching ranges []; retrying
I220116 08:29:59.989307 1314 1@util/log/event_log.go:32  [n1] 104 ={"Timestamp":1642321799989304900,"EventType":"node_join","NodeID":1,"StartedAt":1642321799926364070,"LastUp":1642321799926364070}
I220116 08:29:59.989361 1314 sql/sqlliveness/slinstance/slinstance.go:300  [n1] 105  starting SQL liveness instance
I220116 08:29:59.990849 1542 sql/sqlliveness/slstorage/slstorage.go:406  [n1] 106  inserted sqlliveness session 0e74346e308a4a72b86507536de369d1
I220116 08:29:59.990882 1542 sql/sqlliveness/slinstance/slinstance.go:187  [n1] 107  created new SQL liveness session 0e74346e308a4a72b86507536de369d1
I220116 08:29:59.991069 1650 sql/temporary_schema.go:515  [n1] 108  running temporary object cleanup background job
I220116 08:29:59.991246 1652 sql/sqlstats/persistedsqlstats/provider.go:137  [n1] 109  starting sql-stats-worker with initial delay: 58m21.026762564s
I220116 08:29:59.991441 1650 sql/temporary_schema.go:559  [n1] 110  found 0 temporary schemas
I220116 08:29:59.991464 1650 sql/temporary_schema.go:562  [n1] 111  early exiting temporary schema cleaner as no temporary schemas were found
I220116 08:29:59.991479 1650 sql/temporary_schema.go:563  [n1] 112  completed temporary object cleanup job
I220116 08:29:59.991537 1650 sql/temporary_schema.go:646  [n1] 113  temporary object cleaner next scheduled to run at 2022-01-16 08:59:59.991053099 +0000 UTC
I220116 08:29:59.997280 1692 kv/kvserver/replica_rangefeed.go:651  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 114  RangeFeed closed timestamp is empty
I220116 08:30:00.003275 1402 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 115 ={"Timestamp":1642321799998072675,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I220116 08:30:00.006258 1009 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 116  no need to migrate, cluster already at newest version
I220116 08:30:00.007596 1009 util/log/event_log.go:32  [n1,intExec=set-setting] 117 ={"Timestamp":1642321800005229894,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.2-36'"],"SettingName":"version","Value":"21.2-36"}
I220116 08:30:00.019510 2012 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 118 ={"Timestamp":1642321800016438040,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"b62af44e-4968-4daa-b3ff-191262f66b9d"}
I220116 08:30:00.023515 1877 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 119 ={"Timestamp":1642321800020670344,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I220116 08:30:00.026238 2183 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 120 ={"Timestamp":1642321800023701210,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I220116 08:30:00.030927 1435 3@vendor/github.com/cockroachdb/pebble/event.go:677  [n1,pebble,s1] 121  [JOB 3] WAL created 000004
I220116 08:30:00.042459 1314 server/server_sql.go:1166  [n1] 122  done ensuring all necessary startup migrations have run
I220116 08:30:00.042730 2204 jobs/job_scheduler.go:357  [n1] 123  waiting 4m0s before scheduled jobs daemon start
I220116 08:30:00.044073 2206 server/auto_upgrade.go:55  [n1] 124  no need to upgrade, cluster already at the newest version
I220116 08:30:00.055090 1314 1@server/server.go:2448  [n1] 125  serving sql connections
I220116 08:30:00.055623 1314 testutils/testcluster/testcluster.go:1158  [-] 126  WaitForFullReplication
I220116 08:30:00.055651 1314 testutils/testcluster/testcluster.go:1162  [-] 127  WaitForFullReplication took: 273ns
I220116 08:30:00.055692 1314 2@rpc/context.go:1179  [n1] 128  dialing n1: 127.0.0.1:37681 (default)
--- FAIL: TestRefresh (0.26s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:147: 
        	Error Trace:	cache_test.go:471
        	            				cache_test.go:147
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9375285?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9375285?buildTab=artifacts#/) on master @ [99102ddf4b7602788b422366f1acc14b81c64d24](https://github.com/cockroachdb/cockroach/commits/99102ddf4b7602788b422366f1acc14b81c64d24):


```
I230401 08:52:21.978840 4514 jobs/adopt.go:261  [T1,n1] 352  job 852848613057429505: resuming execution
I230401 08:52:22.035508 4057 jobs/registry.go:1522  [T1,n1] 353  MIGRATION job 852848613057429505: stepping through state running
I230401 08:52:22.090198 4057 jobs/registry.go:1522  [T1,n1] 354  MIGRATION job 852848613057429505: stepping through state succeeded
I230401 08:52:22.170569 2175 jobs/wait.go:145  [T1,n1] 355  waited for 1 [852848613057429505] queued jobs to complete 191.829976ms
I230401 08:52:22.172181 2175 upgrade/upgrademanager/manager.go:270  [T1,n1] 356  running permanent upgrade for version 1000022.2-96
I230401 08:52:22.321225 2175 jobs/registry.go:617  [T1,n1] 357  active version is 1000023.1-4
I230401 08:52:22.366362 2175 upgrade/upgrademanager/manager.go:708  [T1,n1] 358  running Upgrade to 1000022.2-96: "change TTL for SQL Stats system tables"
I230401 08:52:22.366681 4417 jobs/adopt.go:261  [T1,n1] 359  job 852848614351831041: resuming execution
I230401 08:52:22.422616 4626 jobs/registry.go:1522  [T1,n1] 360  MIGRATION job 852848614351831041: stepping through state running
I230401 08:52:22.496485 4537 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=852848614351831041,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 361 ={"Timestamp":1680339142470377121,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230401 08:52:22.523953 4643 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=852848614351831041,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 362 ={"Timestamp":1680339142498306766,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230401 08:52:22.538420 4552 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=852848614351831041,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 363 ={"Timestamp":1680339142525859423,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230401 08:52:22.552921 4661 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=852848614351831041,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 364 ={"Timestamp":1680339142540089699,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230401 08:52:22.560423 4626 jobs/registry.go:1522  [T1,n1] 365  MIGRATION job 852848614351831041: stepping through state succeeded
I230401 08:52:22.638185 2175 jobs/wait.go:145  [T1,n1] 366  waited for 1 [852848614351831041] queued jobs to complete 271.603491ms
I230401 08:52:22.638407 2175 server/server_sql.go:1587  [T1,n1] 367  done ensuring all necessary startup migrations have run
I230401 08:52:22.639692 4669 jobs/job_scheduler.go:407  [T1,n1] 368  waiting 2m0s before scheduled jobs daemon start
I230401 08:52:22.642366 4679 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 369  tenant-settings-watcher: established range feed cache
I230401 08:52:22.644258 4625 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 370  tenant-capability-watcher: established range feed cache
I230401 08:52:22.652325 4682 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 371  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.654261 4625 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 372  received results of a full table scan for tenant capabilities
I230401 08:52:22.655784 4496 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 373  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.659732 4143 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 374  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.661806 4143 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 375  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.673136 4677 server/auto_upgrade.go:78  [T1,n1] 376  no need to upgrade, cluster already at the newest version
I230401 08:52:22.679660 4573 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 377  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.688735 4671 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 378  warmed privileges for virtual tables in 49.071757ms
I230401 08:52:22.747327 4711 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 379  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:22.873465 4715 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 380  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230401 08:52:23.552998 2804 3@pebble/event.go:717  [n1,s1,pebble] 381  [JOB 7] WAL created 000006
I230401 08:52:23.572433 2175 1@server/server_sql.go:1674  [T1,n1] 382  serving sql connections
--- FAIL: TestRefresh (6.75s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
=== RUN   TestRefresh/already_up-to-date
    cache_test.go:180: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:180
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestRefresh/already_up-to-date
    --- FAIL: TestRefresh/already_up-to-date (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9385185?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9385185?buildTab=artifacts#/) on master @ [1e27c65fb5a3810a5e2285cf07b45d95d401cca6](https://github.com/cockroachdb/cockroach/commits/1e27c65fb5a3810a5e2285cf07b45d95d401cca6):


```
I230402 09:02:46.720218 2514 upgrade/upgrademanager/manager.go:270  [T1,n1] 339  running permanent upgrade for version 1000022.2-90
I230402 09:02:46.847450 2514 jobs/registry.go:617  [T1,n1] 340  active version is 1000023.1-4
I230402 09:02:46.888241 2514 upgrade/upgrademanager/manager.go:708  [T1,n1] 341  running Upgrade to 1000022.2-90: "create auto config runner job"
I230402 09:02:46.888552 4190 jobs/adopt.go:261  [T1,n1] 342  job 853133776389406721: resuming execution
I230402 09:02:46.938339 4233 jobs/registry.go:1522  [T1,n1] 343  MIGRATION job 853133776389406721: stepping through state running
I230402 09:02:46.987891 4233 jobs/registry.go:617  [T1,n1,job=MIGRATION id=853133776389406721,upgrade=1000022.2-90] 344  active version is 1000023.1-4
I230402 09:02:47.028013 4225 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=853133776389406721,upgrade=1000022.2-90] 345  job 102: resuming execution
I230402 09:02:47.033993 4233 jobs/registry.go:1522  [T1,n1] 346  MIGRATION job 853133776389406721: stepping through state succeeded
I230402 09:02:47.079901 4277 jobs/registry.go:1522  [T1,n1] 347  AUTO CONFIG RUNNER job 102: stepping through state running
I230402 09:02:47.109660 2514 jobs/wait.go:145  [T1,n1] 348  waited for 1 [853133776389406721] queued jobs to complete 221.168398ms
I230402 09:02:47.110624 2514 upgrade/upgrademanager/manager.go:270  [T1,n1] 349  running permanent upgrade for version 1000022.2-94
I230402 09:02:47.242868 2514 jobs/registry.go:617  [T1,n1] 350  active version is 1000023.1-4
I230402 09:02:47.283553 2514 upgrade/upgrademanager/manager.go:708  [T1,n1] 351  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230402 09:02:47.283947 4302 jobs/adopt.go:261  [T1,n1] 352  job 853133777668767745: resuming execution
I230402 09:02:47.335675 4330 jobs/registry.go:1522  [T1,n1] 353  MIGRATION job 853133777668767745: stepping through state running
I230402 09:02:47.387100 4330 jobs/registry.go:1522  [T1,n1] 354  MIGRATION job 853133777668767745: stepping through state succeeded
I230402 09:02:47.459545 2514 jobs/wait.go:145  [T1,n1] 355  waited for 1 [853133777668767745] queued jobs to complete 175.699375ms
I230402 09:02:47.460511 2514 upgrade/upgrademanager/manager.go:270  [T1,n1] 356  running permanent upgrade for version 1000022.2-96
I230402 09:02:47.604269 2514 jobs/registry.go:617  [T1,n1] 357  active version is 1000023.1-4
I230402 09:02:47.646960 2514 upgrade/upgrademanager/manager.go:708  [T1,n1] 358  running Upgrade to 1000022.2-96: "change TTL for SQL Stats system tables"
I230402 09:02:47.647363 4435 jobs/adopt.go:261  [T1,n1] 359  job 853133778815352833: resuming execution
I230402 09:02:47.699512 4439 jobs/registry.go:1522  [T1,n1] 360  MIGRATION job 853133778815352833: stepping through state running
I230402 09:02:47.769285 4460 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853133778815352833,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 361 ={"Timestamp":1680426167742541124,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 09:02:47.795789 4474 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853133778815352833,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 362 ={"Timestamp":1680426167771504574,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 09:02:47.810213 4395 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853133778815352833,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 363 ={"Timestamp":1680426167797554839,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 09:02:47.824059 4500 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853133778815352833,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 364 ={"Timestamp":1680426167811922323,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 09:02:47.829998 4439 jobs/registry.go:1522  [T1,n1] 365  MIGRATION job 853133778815352833: stepping through state succeeded
I230402 09:02:47.902241 2514 jobs/wait.go:145  [T1,n1] 366  waited for 1 [853133778815352833] queued jobs to complete 254.96739ms
I230402 09:02:47.902464 2514 server/server_sql.go:1587  [T1,n1] 367  done ensuring all necessary startup migrations have run
I230402 09:02:47.904168 4383 jobs/job_scheduler.go:407  [T1,n1] 368  waiting 3m0s before scheduled jobs daemon start
I230402 09:02:47.906864 4479 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 369  tenant-settings-watcher: established range feed cache
I230402 09:02:47.909287 4562 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 370  tenant-capability-watcher: established range feed cache
I230402 09:02:47.916041 4562 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 371  received results of a full table scan for tenant capabilities
I230402 09:02:47.926869 4477 server/auto_upgrade.go:78  [T1,n1] 372  no need to upgrade, cluster already at the newest version
I230402 09:02:47.941893 4385 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 373  warmed privileges for virtual tables in 37.611502ms
I230402 09:02:48.461617 4042 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 374  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 09:02:48.463624 4043 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 375  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 09:02:48.504242 4045 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 376  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 09:02:48.992660 2781 3@pebble/event.go:717  [n1,s1,pebble] 377  [JOB 7] WAL created 000006
I230402 09:02:49.013454 2514 1@server/server_sql.go:1674  [T1,n1] 378  serving sql connections
--- FAIL: TestRefresh (6.42s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9396106?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9396106?buildTab=artifacts#/) on master @ [bebd2dd156778ba666593115569f55e891ec23a7](https://github.com/cockroachdb/cockroach/commits/bebd2dd156778ba666593115569f55e891ec23a7):


```
I230403 09:27:06.686774 2128 upgrade/upgrademanager/manager.go:270  [T1,n1] 340  running permanent upgrade for version 1000022.2-90
I230403 09:27:06.809500 2128 jobs/registry.go:617  [T1,n1] 341  active version is 1000023.1-4
I230403 09:27:06.849233 2128 upgrade/upgrademanager/manager.go:708  [T1,n1] 342  running Upgrade to 1000022.2-90: "create auto config runner job"
I230403 09:27:06.849559 4046 jobs/adopt.go:261  [T1,n1] 343  job 853421675927928833: resuming execution
I230403 09:27:06.897273 4167 jobs/registry.go:1522  [T1,n1] 344  MIGRATION job 853421675927928833: stepping through state running
I230403 09:27:06.943040 4167 jobs/registry.go:617  [T1,n1,job=MIGRATION id=853421675927928833,upgrade=1000022.2-90] 345  active version is 1000023.1-4
I230403 09:27:06.980898 4232 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=853421675927928833,upgrade=1000022.2-90] 346  job 102: resuming execution
I230403 09:27:06.985306 4167 jobs/registry.go:1522  [T1,n1] 347  MIGRATION job 853421675927928833: stepping through state succeeded
I230403 09:27:07.038347 4264 jobs/registry.go:1522  [T1,n1] 348  AUTO CONFIG RUNNER job 102: stepping through state running
I230403 09:27:07.063971 2128 jobs/wait.go:145  [T1,n1] 349  waited for 1 [853421675927928833] queued jobs to complete 214.487978ms
I230403 09:27:07.065069 2128 upgrade/upgrademanager/manager.go:270  [T1,n1] 350  running permanent upgrade for version 1000022.2-94
I230403 09:27:07.201262 2128 jobs/registry.go:617  [T1,n1] 351  active version is 1000023.1-4
I230403 09:27:07.244053 2128 upgrade/upgrademanager/manager.go:708  [T1,n1] 352  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230403 09:27:07.244428 4185 jobs/adopt.go:261  [T1,n1] 353  job 853421677167542273: resuming execution
I230403 09:27:07.296411 4081 jobs/registry.go:1522  [T1,n1] 354  MIGRATION job 853421677167542273: stepping through state running
I230403 09:27:07.342715 4081 jobs/registry.go:1522  [T1,n1] 355  MIGRATION job 853421677167542273: stepping through state succeeded
I230403 09:27:07.412381 2128 jobs/wait.go:145  [T1,n1] 356  waited for 1 [853421677167542273] queued jobs to complete 168.066427ms
I230403 09:27:07.413164 2128 upgrade/upgrademanager/manager.go:270  [T1,n1] 357  running permanent upgrade for version 1000022.2-96
I230403 09:27:07.546158 2128 jobs/registry.go:617  [T1,n1] 358  active version is 1000023.1-4
I230403 09:27:07.589067 2128 upgrade/upgrademanager/manager.go:708  [T1,n1] 359  running Upgrade to 1000022.2-96: "change TTL for SQL Stats system tables"
I230403 09:27:07.589355 4383 jobs/adopt.go:261  [T1,n1] 360  job 853421678308196353: resuming execution
I230403 09:27:07.638404 4418 jobs/registry.go:1522  [T1,n1] 361  MIGRATION job 853421678308196353: stepping through state running
I230403 09:27:07.704891 4424 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853421678308196353,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 362 ={"Timestamp":1680514027680258726,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230403 09:27:07.730108 3344 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853421678308196353,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 363 ={"Timestamp":1680514027706626867,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230403 09:27:07.743769 4440 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853421678308196353,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 364 ={"Timestamp":1680514027731634136,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230403 09:27:07.757510 4457 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853421678308196353,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 365 ={"Timestamp":1680514027745342328,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230403 09:27:07.763567 4418 jobs/registry.go:1522  [T1,n1] 366  MIGRATION job 853421678308196353: stepping through state succeeded
I230403 09:27:07.834627 2128 jobs/wait.go:145  [T1,n1] 367  waited for 1 [853421678308196353] queued jobs to complete 245.326066ms
I230403 09:27:07.834827 2128 server/server_sql.go:1587  [T1,n1] 368  done ensuring all necessary startup migrations have run
I230403 09:27:07.836199 4488 jobs/job_scheduler.go:407  [T1,n1] 369  waiting 2m0s before scheduled jobs daemon start
I230403 09:27:07.838766 4494 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 370  tenant-settings-watcher: established range feed cache
I230403 09:27:07.840885 4416 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 371  tenant-capability-watcher: established range feed cache
I230403 09:27:07.846670 4416 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 372  received results of a full table scan for tenant capabilities
I230403 09:27:07.862271 4492 server/auto_upgrade.go:78  [T1,n1] 373  no need to upgrade, cluster already at the newest version
I230403 09:27:07.873655 4490 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 374  warmed privileges for virtual tables in 37.230705ms
I230403 09:27:08.508279 4760 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 375  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230403 09:27:08.510174 4399 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 376  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230403 09:27:08.552794 4433 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 377  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230403 09:27:08.845876 2769 3@pebble/event.go:717  [n1,s1,pebble] 378  [JOB 7] WAL created 000006
I230403 09:27:08.871597 2128 1@server/server_sql.go:1674  [T1,n1] 379  serving sql connections
--- FAIL: TestRefresh (6.28s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9484394?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9484394?buildTab=artifacts#/) on master @ [8e6f530644eb35313a479af529b1f672c12dca38](https://github.com/cockroachdb/cockroach/commits/8e6f530644eb35313a479af529b1f672c12dca38):


```
I230407 08:52:07.597963 498993 upgrade/upgrademanager/manager.go:742  [T1,n1] 432  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230407 08:52:07.598288 888423 jobs/adopt.go:261  [T1,n1] 433  job 854547259136409601: resuming execution
I230407 08:52:07.667024 890454 jobs/registry.go:1538  [T1,n1] 434  MIGRATION job 854547259136409601: stepping through state running
I230407 08:52:07.731182 890454 jobs/registry.go:1538  [T1,n1] 435  MIGRATION job 854547259136409601: stepping through state succeeded
I230407 08:52:07.824234 498993 jobs/wait.go:145  [T1,n1] 436  waited for 1 [854547259136409601] queued jobs to complete 226.003899ms
I230407 08:52:07.825603 498993 upgrade/upgrademanager/manager.go:283  [T1,n1] 437  running permanent upgrade for version 1000022.2-98
I230407 08:52:07.940294 498993 jobs/registry.go:608  [T1,n1] 438  active version is 1000023.1-2
I230407 08:52:08.000559 498993 upgrade/upgrademanager/manager.go:742  [T1,n1] 439  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230407 08:52:08.000852 911171 jobs/adopt.go:261  [T1,n1] 440  job 854547260459155457: resuming execution
I230407 08:52:08.079326 913367 jobs/registry.go:1538  [T1,n1] 441  MIGRATION job 854547260459155457: stepping through state running
I230407 08:52:08.160978 919003 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854547260459155457,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 442 ={"Timestamp":1680857528141754492,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230407 08:52:08.180812 920414 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854547260459155457,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 443 ={"Timestamp":1680857528162656371,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230407 08:52:08.198128 922656 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854547260459155457,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 444 ={"Timestamp":1680857528182534107,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230407 08:52:08.217895 924937 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854547260459155457,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 445 ={"Timestamp":1680857528199736901,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230407 08:52:08.227837 913367 jobs/registry.go:1538  [T1,n1] 446  MIGRATION job 854547260459155457: stepping through state succeeded
I230407 08:52:08.340850 498993 jobs/wait.go:145  [T1,n1] 447  waited for 1 [854547260459155457] queued jobs to complete 340.051016ms
I230407 08:52:08.341148 498993 server/server_sql.go:1626  [T1,n1] 448  done ensuring all necessary startup migrations have run
I230407 08:52:08.344024 934562 jobs/job_scheduler.go:407  [T1,n1] 449  waiting 3m0s before scheduled jobs daemon start
I230407 08:52:08.348707 935294 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 450  tenant-settings-watcher: established range feed cache
I230407 08:52:08.353926 936107 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 451  tenant-capability-watcher: established range feed cache
I230407 08:52:08.361755 936107 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 452  received results of a full table scan for tenant capabilities
I230407 08:52:08.385333 933433 server/auto_upgrade.go:78  [T1,n1] 453  no need to upgrade, cluster already at the newest version
I230407 08:52:08.394216 934572 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 454  warmed privileges for virtual tables in 49.722401ms
I230407 08:52:08.443677 507469 3@pebble/event.go:717  [n1,s1,pebble] 455  [JOB 7] WAL created 000006
I230407 08:52:08.448392 498993 1@server/server_sql.go:1744  [T1,n1] 456  serving sql connections
--- FAIL: TestRefresh (6.81s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230407 08:52:08.558019 949296 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.559650 949663 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.561535 949706 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.563333 949706 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.565105 950844 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.566727 951278 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.568652 951567 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.570341 952084 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.572291 949887 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.573980 952705 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.575290 953240 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.576670 953527 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.578401 953924 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.580669 954277 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230407 08:52:08.582606 954728 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9501290?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9501290?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
I230408 07:46:06.617675 858023 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=854817394786893825,upgrade=1000022.2-90] 382  job 102: resuming execution
I230408 07:46:06.625658 848508 jobs/registry.go:1538  [T1,n1] 383  MIGRATION job 854817394786893825: stepping through state succeeded
I230408 07:46:06.701642 861992 jobs/registry.go:1538  [T1,n1] 384  AUTO CONFIG RUNNER job 102: stepping through state running
I230408 07:46:06.732142 491500 jobs/wait.go:145  [T1,n1] 385  waited for 1 [854817394786893825] queued jobs to complete 295.899738ms
I230408 07:46:06.733960 491500 upgrade/upgrademanager/manager.go:283  [T1,n1] 386  running permanent upgrade for version 1000022.2-94
I230408 07:46:06.756420 861992 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 387  using provider with type acprovider.NoTaskProvider
I230408 07:46:06.756517 861992 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 388  waiting for environment activation...
I230408 07:46:06.756629 861992 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 389  waiting for environment activation...
I230408 07:46:06.861338 491500 jobs/registry.go:608  [T1,n1] 390  active version is 1000023.1-2
I230408 07:46:06.930651 491500 upgrade/upgrademanager/manager.go:742  [T1,n1] 391  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230408 07:46:06.930871 881094 jobs/adopt.go:261  [T1,n1] 392  job 854817396274167809: resuming execution
I230408 07:46:07.009657 883381 jobs/registry.go:1538  [T1,n1] 393  MIGRATION job 854817396274167809: stepping through state running
I230408 07:46:07.067461 500277 kv/kvserver/replica_rangefeed.go:714  [T1,n1,s1,r22/1:/Table/2{0-1}] 394  RangeFeed closed timestamp is empty
I230408 07:46:07.076941 883381 jobs/registry.go:1538  [T1,n1] 395  MIGRATION job 854817396274167809: stepping through state succeeded
I230408 07:46:07.165707 491500 jobs/wait.go:145  [T1,n1] 396  waited for 1 [854817396274167809] queued jobs to complete 234.923717ms
I230408 07:46:07.166858 491500 upgrade/upgrademanager/manager.go:283  [T1,n1] 397  running permanent upgrade for version 1000022.2-98
I230408 07:46:07.272352 491500 jobs/registry.go:608  [T1,n1] 398  active version is 1000023.1-2
I230408 07:46:07.330770 491500 upgrade/upgrademanager/manager.go:742  [T1,n1] 399  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230408 07:46:07.330986 904062 jobs/adopt.go:261  [T1,n1] 400  job 854817397692661761: resuming execution
I230408 07:46:07.402512 906226 jobs/registry.go:1538  [T1,n1] 401  MIGRATION job 854817397692661761: stepping through state running
I230408 07:46:07.485407 911963 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854817397692661761,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 402 ={"Timestamp":1680939967466634846,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230408 07:46:07.507324 913432 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854817397692661761,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 403 ={"Timestamp":1680939967487436258,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230408 07:46:07.523488 915741 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854817397692661761,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 404 ={"Timestamp":1680939967508638187,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230408 07:46:07.541947 917875 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=854817397692661761,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 405 ={"Timestamp":1680939967525353482,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230408 07:46:07.550446 906226 jobs/registry.go:1538  [T1,n1] 406  MIGRATION job 854817397692661761: stepping through state succeeded
I230408 07:46:07.645067 491500 jobs/wait.go:145  [T1,n1] 407  waited for 1 [854817397692661761] queued jobs to complete 314.128233ms
I230408 07:46:07.645150 491500 server/server_sql.go:1618  [T1,n1] 408  done ensuring all necessary startup migrations have run
I230408 07:46:07.646996 927508 jobs/job_scheduler.go:407  [T1,n1] 409  waiting 2m0s before scheduled jobs daemon start
I230408 07:46:07.651208 928290 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 410  tenant-settings-watcher: established range feed cache
I230408 07:46:07.655538 929035 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 411  tenant-capability-watcher: established range feed cache
I230408 07:46:07.660476 929035 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 412  received results of a full table scan for tenant capabilities
I230408 07:46:07.680043 927535 server/auto_upgrade.go:78  [T1,n1] 413  no need to upgrade, cluster already at the newest version
I230408 07:46:07.688036 927526 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 414  warmed privileges for virtual tables in 40.891166ms
I230408 07:46:07.740380 500080 3@pebble/event.go:717  [n1,s1,pebble] 415  [JOB 7] WAL created 000006
I230408 07:46:07.745168 491500 1@server/server_sql.go:1736  [T1,n1] 416  serving sql connections
--- FAIL: TestRefresh (6.55s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230408 07:46:07.881619 942480 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 417  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230408 07:46:07.887289 943169 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 418  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230408 07:46:07.890764 943134 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 419  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230408 07:46:07.892286 943134 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 420  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230408 07:46:07.894335 944944 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 421  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9509499?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9509499?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
I230409 09:14:08.815955 936153 jobs/job_scheduler.go:407  [T1,n1] 539  waiting 4m0s before scheduled jobs daemon start
I230409 09:14:08.818088 936539 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-boundaries-watcher] 540  Establishing MuxRangeFeed (/Table/5{7-8}@1681031648.816423697,0...; 1 spans)
I230409 09:14:08.819883 936877 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 541  tenant-settings-watcher: established range feed cache
I230409 09:14:08.820896 936555 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-boundaries-watcher] 542  MuxRangeFeed starting for range /Table/5{7-8}@1681031648.816423697,0 (rangeID 58, attempt 0)
I230409 09:14:08.821230 937087 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 543  Establishing MuxRangeFeed to node 1
I230409 09:14:08.824428 937798 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 544  tenant-capability-watcher: established range feed cache
I230409 09:14:08.829745 937798 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 545  received results of a full table scan for tenant capabilities
I230409 09:14:08.852820 936259 server/auto_upgrade.go:78  [T1,n1] 546  no need to upgrade, cluster already at the newest version
I230409 09:14:08.861980 936179 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 547  warmed privileges for virtual tables in 45.948846ms
I230409 09:14:08.877472 944681 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 548  Establishing MuxRangeFeed (/Table/8/{1-2}@1681031648.823991212,0...; 1 spans)
I230409 09:14:08.877705 944930 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 549  MuxRangeFeed starting for range /Table/8/{1-2}@1681031648.823991212,0 (rangeID 12, attempt 0)
I230409 09:14:08.877866 944940 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 550  Establishing MuxRangeFeed to node 1
I230409 09:14:08.879333 945021 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 551  Establishing MuxRangeFeed (/Table/5{0-1}@1681031648.819698896,0...; 1 spans)
I230409 09:14:08.879590 945078 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 552  MuxRangeFeed starting for range /Table/5{0-1}@1681031648.819698896,0 (rangeID 51, attempt 0)
I230409 09:14:08.879816 944991 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 553  Establishing MuxRangeFeed to node 1
I230409 09:14:08.928740 507607 3@pebble/event.go:717  [n1,s1,pebble] 554  [JOB 7] WAL created 000006
I230409 09:14:08.932778 499161 1@server/server_sql.go:1736  [T1,n1] 555  serving sql connections
--- FAIL: TestRefresh (6.77s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230409 09:14:09.026672 950991 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 556  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.028636 951394 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 557  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.030366 951293 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 558  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.031794 951293 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 559  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.033136 952550 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 560  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.035177 952857 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 561  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.036343 953201 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 562  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.038121 953579 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 563  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.039707 953904 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 564  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.041128 954372 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 565  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.042461 954711 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 566  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.043992 955004 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.045335 955368 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.046940 955740 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.048516 956110 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.050164 956462 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.051683 956838 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.053077 957151 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.054712 957518 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.055969 957954 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.057653 958140 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.059061 958630 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230409 09:14:09.060411 958921 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9518516?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9518516?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
I230410 09:25:15.463536 868999 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 504  waiting for environment activation...
I230410 09:25:15.565729 495711 jobs/registry.go:608  [T1,n1] 505  active version is 1000023.1-2
I230410 09:25:15.626453 495711 upgrade/upgrademanager/manager.go:742  [T1,n1] 506  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230410 09:25:15.626748 886778 jobs/adopt.go:261  [T1,n1] 507  job 855403119998599169: resuming execution
I230410 09:25:15.702225 888843 jobs/registry.go:1538  [T1,n1] 508  MIGRATION job 855403119998599169: stepping through state running
I230410 09:25:15.768956 888843 jobs/registry.go:1538  [T1,n1] 509  MIGRATION job 855403119998599169: stepping through state succeeded
I230410 09:25:15.865122 495711 jobs/wait.go:145  [T1,n1] 510  waited for 1 [855403119998599169] queued jobs to complete 238.427707ms
I230410 09:25:15.866617 495711 upgrade/upgrademanager/manager.go:283  [T1,n1] 511  running permanent upgrade for version 1000022.2-98
I230410 09:25:15.992576 495711 jobs/registry.go:608  [T1,n1] 512  active version is 1000023.1-2
I230410 09:25:16.077388 495711 upgrade/upgrademanager/manager.go:742  [T1,n1] 513  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230410 09:25:16.077680 910939 jobs/adopt.go:261  [T1,n1] 514  job 855403121432035329: resuming execution
I230410 09:25:16.154085 913161 jobs/registry.go:1538  [T1,n1] 515  MIGRATION job 855403121432035329: stepping through state running
I230410 09:25:16.229377 918826 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855403121432035329,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 516 ={"Timestamp":1681118716212248178,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230410 09:25:16.252477 920139 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855403121432035329,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 517 ={"Timestamp":1681118716230998507,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230410 09:25:16.275917 922653 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855403121432035329,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 518 ={"Timestamp":1681118716255414544,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230410 09:25:16.295925 924822 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855403121432035329,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 519 ={"Timestamp":1681118716277915967,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230410 09:25:16.305560 913161 jobs/registry.go:1538  [T1,n1] 520  MIGRATION job 855403121432035329: stepping through state succeeded
I230410 09:25:16.408936 495711 jobs/wait.go:145  [T1,n1] 521  waited for 1 [855403121432035329] queued jobs to complete 331.331676ms
I230410 09:25:16.409127 495711 server/server_sql.go:1618  [T1,n1] 522  done ensuring all necessary startup migrations have run
I230410 09:25:16.411427 934231 jobs/job_scheduler.go:407  [T1,n1] 523  waiting 3m0s before scheduled jobs daemon start
I230410 09:25:16.414086 934623 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-boundaries-watcher] 524  Establishing MuxRangeFeed (/Table/5{7-8}@1681118716.412082940,0...; 1 spans)
I230410 09:25:16.416325 935060 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 525  tenant-settings-watcher: established range feed cache
I230410 09:25:16.416621 934667 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-boundaries-watcher] 526  MuxRangeFeed starting for range /Table/5{7-8}@1681118716.412082940,0 (rangeID 58, attempt 0)
I230410 09:25:16.416894 935189 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 527  Establishing MuxRangeFeed to node 1
I230410 09:25:16.421354 935845 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 528  tenant-capability-watcher: established range feed cache
I230410 09:25:16.425761 935845 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 529  received results of a full table scan for tenant capabilities
I230410 09:25:16.445487 934344 server/auto_upgrade.go:78  [T1,n1] 530  no need to upgrade, cluster already at the newest version
I230410 09:25:16.454011 934241 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 531  warmed privileges for virtual tables in 42.540231ms
I230410 09:25:16.472332 941699 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 532  Establishing MuxRangeFeed (/Table/5{0-1}@1681118716.416041206,0...; 1 spans)
I230410 09:25:16.472591 941708 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 533  MuxRangeFeed starting for range /Table/5{0-1}@1681118716.416041206,0 (rangeID 51, attempt 0)
I230410 09:25:16.472773 941718 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 534  Establishing MuxRangeFeed to node 1
I230410 09:25:16.478551 941814 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 535  Establishing MuxRangeFeed (/Table/8/{1-2}@1681118716.420982420,0...; 1 spans)
I230410 09:25:16.478775 941823 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 536  MuxRangeFeed starting for range /Table/8/{1-2}@1681118716.420982420,0 (rangeID 12, attempt 0)
I230410 09:25:16.479018 941913 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 537  Establishing MuxRangeFeed to node 1
I230410 09:25:16.524964 504233 3@pebble/event.go:717  [n1,s1,pebble] 538  [JOB 7] WAL created 000006
I230410 09:25:16.530052 495711 1@server/server_sql.go:1736  [T1,n1] 539  serving sql connections
--- FAIL: TestRefresh (7.18s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230410 09:25:16.645339 948829 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 540  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230410 09:25:16.647199 949159 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 541  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230410 09:25:16.648674 949176 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 542  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230410 09:25:16.649922 949176 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 543  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556783?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9556783?buildTab=artifacts#/) on master @ [582880ae2bb5d08e83a9871735deeeb3ed892e7d](https://github.com/cockroachdb/cockroach/commits/582880ae2bb5d08e83a9871735deeeb3ed892e7d):


```
I230412 10:18:31.980462 484790 jobs/registry.go:608  [T1,n1] 368  active version is 1000023.1-2
I230412 10:18:32.041310 484790 upgrade/upgrademanager/manager.go:742  [T1,n1] 369  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230412 10:18:32.041491 880658 jobs/adopt.go:261  [T1,n1] 370  job 855979825008738305: resuming execution
I230412 10:18:32.129981 883047 jobs/registry.go:1538  [T1,n1] 371  MIGRATION job 855979825008738305: stepping through state running
I230412 10:18:32.200271 883047 jobs/registry.go:1538  [T1,n1] 372  MIGRATION job 855979825008738305: stepping through state succeeded
I230412 10:18:32.302132 484790 jobs/wait.go:145  [T1,n1] 373  waited for 1 [855979825008738305] queued jobs to complete 260.673614ms
I230412 10:18:32.303937 484790 upgrade/upgrademanager/manager.go:283  [T1,n1] 374  running permanent upgrade for version 1000022.2-98
I230412 10:18:32.425045 484790 jobs/registry.go:608  [T1,n1] 375  active version is 1000023.1-2
I230412 10:18:32.488834 484790 upgrade/upgrademanager/manager.go:742  [T1,n1] 376  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230412 10:18:32.489101 905977 jobs/adopt.go:261  [T1,n1] 377  job 855979826557911041: resuming execution
I230412 10:18:32.560102 908376 jobs/registry.go:1538  [T1,n1] 378  MIGRATION job 855979826557911041: stepping through state running
I230412 10:18:32.638476 914009 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855979826557911041,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 379 ={"Timestamp":1681294712620136417,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230412 10:18:32.660760 915375 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855979826557911041,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 380 ={"Timestamp":1681294712639980987,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230412 10:18:32.681941 917803 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855979826557911041,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 381 ={"Timestamp":1681294712663350948,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230412 10:18:32.703114 919972 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855979826557911041,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 382 ={"Timestamp":1681294712684004248,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230412 10:18:32.713404 908376 jobs/registry.go:1538  [T1,n1] 383  MIGRATION job 855979826557911041: stepping through state succeeded
I230412 10:18:32.819434 484790 jobs/wait.go:145  [T1,n1] 384  waited for 1 [855979826557911041] queued jobs to complete 330.389075ms
I230412 10:18:32.819600 484790 server/server_sql.go:1618  [T1,n1] 385  done ensuring all necessary startup migrations have run
I230412 10:18:32.821019 929335 jobs/job_scheduler.go:407  [T1,n1] 386  waiting 4m0s before scheduled jobs daemon start
I230412 10:18:32.825784 930117 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 387  tenant-settings-watcher: established range feed cache
I230412 10:18:32.829841 930826 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 388  tenant-capability-watcher: established range feed cache
I230412 10:18:32.833272 930826 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 389  received results of a full table scan for tenant capabilities
I230412 10:18:32.860063 929391 server/auto_upgrade.go:78  [T1,n1] 390  no need to upgrade, cluster already at the newest version
I230412 10:18:32.870755 929345 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 391  warmed privileges for virtual tables in 49.626452ms
I230412 10:18:32.915523 493355 3@pebble/event.go:717  [n1,s1,pebble] 392  [JOB 7] WAL created 000006
I230412 10:18:32.920467 484790 1@server/server_sql.go:1736  [T1,n1] 393  serving sql connections
--- FAIL: TestRefresh (7.18s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230412 10:18:33.037086 943177 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 394  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.039110 943993 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 395  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.040660 943951 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 396  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.042031 943951 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 397  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.043939 944797 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 398  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.045396 945312 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 399  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.047070 945698 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 400  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.048552 946099 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 401  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.049938 946501 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 402  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.051638 946653 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 403  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.052978 947172 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 404  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.054671 947471 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 405  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.056240 947883 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 406  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230412 10:18:33.058151 948299 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 407  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9580783?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9580783?buildTab=artifacts#/) on master @ [678cfd4cbebbf3cced16747baf42a4c54cb2a92d](https://github.com/cockroachdb/cockroach/commits/678cfd4cbebbf3cced16747baf42a4c54cb2a92d):


```
I230413 10:55:41.760846 5505 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.762545 5554 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.764360 5555 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.766215 5556 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.767931 5557 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.769686 5558 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.771585 5559 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.773727 5560 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.775551 5561 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.777436 5562 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.779773 5563 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.781133 5564 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.782730 5565 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.784297 5566 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.785845 5567 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.787144 5568 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.788523 5569 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.790285 5570 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.791828 5571 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.793163 5572 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.794848 5573 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.796338 5574 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.798019 5575 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.799974 5576 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.801303 5577 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.802845 5578 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.804781 5579 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.806649 5580 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.808366 5581 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.810352 5582 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.812134 5583 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.813544 5584 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.815409 5585 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.816981 5586 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.818399 5194 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.820270 5195 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.821984 5545 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:41.823780 5546 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230413 10:55:42.277633 3193 3@pebble/event.go:717  [n1,s1,pebble] 507  [JOB 7] WAL created 000006
I230413 10:55:42.297074 2757 1@server/server_sql.go:1736  [T1,n1] 508  serving sql connections
--- FAIL: TestRefresh (6.88s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9602280?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9602280?buildTab=artifacts#/) on master @ [99baf6f17c185d915595f7f1fd8b6985bf26f6f1](https://github.com/cockroachdb/cockroach/commits/99baf6f17c185d915595f7f1fd8b6985bf26f6f1):


```
I230414 11:04:41.305851 5557 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.307517 5558 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.309277 5559 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.311077 5560 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.313007 5561 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.314781 5562 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.316508 5563 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.318393 5564 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.320142 5565 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.322067 5566 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.323924 5567 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.325688 5568 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.327498 5569 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.329269 5570 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.331072 5571 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.332256 5572 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.333424 5573 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.334614 5574 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.335985 5575 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.337368 5576 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.338570 5577 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.340001 5578 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.341168 5579 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.342457 5580 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.343747 5581 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.344957 5582 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.346490 5583 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.348197 5584 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.350055 5585 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.351751 5586 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.353451 5587 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.354790 5588 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.355983 5589 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.357730 5590 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.359222 5591 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.360955 5592 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.363310 5593 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.364654 5594 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230414 11:04:41.794030 3167 3@pebble/event.go:717  [n1,s1,pebble] 508  [JOB 7] WAL created 000006
I230414 11:04:41.823250 2345 1@server/server_sql.go:1736  [T1,n1] 509  serving sql connections
--- FAIL: TestRefresh (6.76s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9619180?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9619180?buildTab=artifacts#/) on master @ [51fcf796c5595549e90fca3cc1fd8165ee14fd6c](https://github.com/cockroachdb/cockroach/commits/51fcf796c5595549e90fca3cc1fd8165ee14fd6c):


```
I230415 10:33:42.136420 5533 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.137723 5534 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.139112 5535 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.140479 5536 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.141898 5537 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.143280 5538 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.144589 5539 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.145950 5540 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.147387 5541 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.148670 5542 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.149998 5543 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.151347 5544 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.152634 5545 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 579  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.154016 5546 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 580  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.155229 5547 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 581  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.156513 5548 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 582  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.158055 5549 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 583  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.159320 5550 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 584  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.161033 5551 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.162605 5552 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.164351 5553 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.165678 5554 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.167181 5555 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.168617 5556 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.170401 5557 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.172598 5558 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.174347 5559 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.175891 5560 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.177396 5561 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.178909 5562 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.180261 5563 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.182223 5564 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.183988 5565 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.186262 5566 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.188674 5567 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.190722 5568 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.192358 5569 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.193707 5570 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230415 10:33:42.672127 3204 3@pebble/event.go:717  [n1,s1,pebble] 605  [JOB 7] WAL created 000006
I230415 10:33:42.693033 2728 1@server/server_sql.go:1736  [T1,n1] 606  serving sql connections
--- FAIL: TestRefresh (7.03s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9630497?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9630497?buildTab=artifacts#/) on master @ [c08946af1f935d543a033de6b7b585c4d65288c4](https://github.com/cockroachdb/cockroach/commits/c08946af1f935d543a033de6b7b585c4d65288c4):


```
I230416 11:24:37.268482 5507 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.269786 5508 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.271020 5509 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.272382 5510 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.273714 5511 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.275020 5512 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.276350 5513 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.277626 5514 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.279032 5515 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.280337 5516 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.281734 5517 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 579  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.282955 5518 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 580  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.284229 5519 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 581  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.285487 5520 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 582  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.286754 5521 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 583  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.288225 3837 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 584  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.289663 3838 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.290939 3839 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.292281 3840 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.293877 5263 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.295310 5264 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.296599 5265 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.297908 5522 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.299225 5523 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.300563 5524 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.302031 5525 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.303446 5526 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.304798 5527 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.306142 5528 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.307603 5529 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.309176 5530 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.310971 5383 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.312683 5384 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.314299 5385 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.315940 5386 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.317561 5387 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.319231 5388 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 605  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.320925 5389 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 606  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230416 11:24:37.800750 3182 3@pebble/event.go:717  [n1,s1,pebble] 607  [JOB 7] WAL created 000006
I230416 11:24:37.820246 1191 1@server/server_sql.go:1736  [T1,n1] 608  serving sql connections
--- FAIL: TestRefresh (6.91s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9641280?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9641280?buildTab=artifacts#/) on master @ [3382b2737ceb72de9b14d1d5923773718be5bca7](https://github.com/cockroachdb/cockroach/commits/3382b2737ceb72de9b14d1d5923773718be5bca7):


```
I230417 10:32:06.368113 477909 jobs/wait.go:145  [T1,n1] 355  waited for 1 [857398069994618881] queued jobs to complete 319.988211ms
I230417 10:32:06.369852 477909 upgrade/upgrademanager/manager.go:283  [T1,n1] 356  running permanent upgrade for version 1000022.2-90
I230417 10:32:06.490905 477909 jobs/registry.go:613  [T1,n1] 357  active version is 1000023.1-2
I230417 10:32:06.498473 486703 kv/kvserver/replica_rangefeed.go:709  [T1,n1,s1,r22/1:/Table/2{0-1}] 358  RangeFeed closed timestamp is empty
I230417 10:32:06.556073 477909 upgrade/upgrademanager/manager.go:742  [T1,n1] 359  running Upgrade to 1000022.2-90: "create auto config runner job"
I230417 10:32:06.556352 846151 jobs/adopt.go:261  [T1,n1] 360  job 857398071689084929: resuming execution
I230417 10:32:06.635611 848212 jobs/registry.go:1543  [T1,n1] 361  MIGRATION job 857398071689084929: stepping through state running
I230417 10:32:06.704180 848212 jobs/registry.go:613  [T1,n1,job=MIGRATION id=857398071689084929,upgrade=1000022.2-90] 362  active version is 1000023.1-2
I230417 10:32:06.765986 857596 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=857398071689084929,upgrade=1000022.2-90] 363  job 102: resuming execution
I230417 10:32:06.774589 848212 jobs/registry.go:1543  [T1,n1] 364  MIGRATION job 857398071689084929: stepping through state succeeded
I230417 10:32:06.863275 862593 jobs/registry.go:1543  [T1,n1] 365  AUTO CONFIG RUNNER job 102: stepping through state running
I230417 10:32:06.902440 477909 jobs/wait.go:145  [T1,n1] 366  waited for 1 [857398071689084929] queued jobs to complete 346.135889ms
I230417 10:32:06.904218 477909 upgrade/upgrademanager/manager.go:283  [T1,n1] 367  running permanent upgrade for version 1000022.2-98
I230417 10:32:06.927610 862593 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 368  using provider with type acprovider.NoTaskProvider
I230417 10:32:06.927710 862593 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 369  waiting for environment activation...
I230417 10:32:06.927801 862593 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 370  waiting for environment activation...
I230417 10:32:07.036398 477909 jobs/registry.go:613  [T1,n1] 371  active version is 1000023.1-2
I230417 10:32:07.097320 477909 upgrade/upgrademanager/manager.go:742  [T1,n1] 372  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230417 10:32:07.097540 880614 jobs/adopt.go:261  [T1,n1] 373  job 857398073440108545: resuming execution
I230417 10:32:07.181613 882781 jobs/registry.go:1543  [T1,n1] 374  MIGRATION job 857398073440108545: stepping through state running
I230417 10:32:07.266268 888589 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857398073440108545,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 375 ={"Timestamp":1681727527247324319,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230417 10:32:07.290479 889807 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857398073440108545,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 376 ={"Timestamp":1681727527267971054,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230417 10:32:07.331648 892122 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857398073440108545,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 377 ={"Timestamp":1681727527294601202,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230417 10:32:07.354581 896490 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857398073440108545,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 378 ={"Timestamp":1681727527334390884,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230417 10:32:07.365976 882781 jobs/registry.go:1543  [T1,n1] 379  MIGRATION job 857398073440108545: stepping through state succeeded
I230417 10:32:07.466311 477909 jobs/wait.go:145  [T1,n1] 380  waited for 1 [857398073440108545] queued jobs to complete 368.815219ms
I230417 10:32:07.466458 477909 server/server_sql.go:1618  [T1,n1] 381  done ensuring all necessary startup migrations have run
I230417 10:32:07.468602 905812 jobs/job_scheduler.go:407  [T1,n1] 382  waiting 4m0s before scheduled jobs daemon start
I230417 10:32:07.473678 906579 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 383  tenant-settings-watcher: established range feed cache
I230417 10:32:07.479828 907362 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 384  tenant-capability-watcher: established range feed cache
I230417 10:32:07.486428 907362 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 385  received results of a full table scan for tenant capabilities
I230417 10:32:07.519618 905897 server/auto_upgrade.go:78  [T1,n1] 386  no need to upgrade, cluster already at the newest version
I230417 10:32:07.528070 905822 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 387  warmed privileges for virtual tables in 59.257608ms
I230417 10:32:07.593475 486415 3@pebble/event.go:717  [n1,s1,pebble] 388  [JOB 7] WAL created 000006
I230417 10:32:07.598068 477909 1@server/server_sql.go:1736  [T1,n1] 389  serving sql connections
--- FAIL: TestRefresh (6.95s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230417 10:32:07.711709 919944 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 390  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:32:07.714539 920388 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 391  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:32:07.716301 920336 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 392  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:32:07.718244 920336 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 393  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:32:07.720391 921500 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 394  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9641284?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9641284?buildTab=artifacts#/) on master @ [3382b2737ceb72de9b14d1d5923773718be5bca7](https://github.com/cockroachdb/cockroach/commits/3382b2737ceb72de9b14d1d5923773718be5bca7):


```
I230417 10:38:13.179675 5513 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.180953 5514 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.182123 5494 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.183902 5495 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.185675 5476 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.187506 5477 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.189374 5478 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.191056 5496 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.192508 5497 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.194075 5498 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.195640 5499 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.197162 5500 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.198660 5501 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.200334 5502 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.201874 5503 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.203601 5504 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.205207 5505 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.206365 5522 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.207791 5515 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.209433 5516 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.211004 5517 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.212508 5518 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.214244 5519 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.216423 5520 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.218331 5521 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.219876 5538 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.221367 5479 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.222608 5480 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.224106 5481 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.225724 5482 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.227150 5483 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.228835 5484 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.230315 5485 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.231903 5486 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.233471 5487 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.234976 5488 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.236405 5489 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.237830 5554 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230417 10:38:13.718713 3141 3@pebble/event.go:717  [n1,s1,pebble] 507  [JOB 7] WAL created 000006
I230417 10:38:13.739589 2841 1@server/server_sql.go:1736  [T1,n1] 508  serving sql connections
--- FAIL: TestRefresh (6.69s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9661632?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9661632?buildTab=artifacts#/) on master @ [1cd507a7c6582fd91bb56123e62bd4fde45c4d22](https://github.com/cockroachdb/cockroach/commits/1cd507a7c6582fd91bb56123e62bd4fde45c4d22):


```
I230418 10:54:53.423195 920546 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 530  Establishing MuxRangeFeed to node 1
I230418 10:54:53.429509 920665 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 531  tenant-capability-watcher: established range feed cache
I230418 10:54:53.433686 920665 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 532  received results of a full table scan for tenant capabilities
I230418 10:54:53.469429 919550 server/auto_upgrade.go:78  [T1,n1] 533  no need to upgrade, cluster already at the newest version
I230418 10:54:53.471853 919079 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 534  warmed privileges for virtual tables in 55.253656ms
I230418 10:54:53.479770 922005 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 535  Establishing MuxRangeFeed (/Table/5{0-1}@1681815293.421517188,2...; 1 spans)
I230418 10:54:53.480128 922014 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 536  MuxRangeFeed starting for range /Table/5{0-1}@1681815293.421517188,2 (rangeID 51, attempt 0)
I230418 10:54:53.480398 926552 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 537  Establishing MuxRangeFeed to node 1
I230418 10:54:53.490439 926732 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 538  Establishing MuxRangeFeed (/Table/8/{1-2}@1681815293.429228883,0...; 1 spans)
I230418 10:54:53.490798 926687 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 539  MuxRangeFeed starting for range /Table/8/{1-2}@1681815293.429228883,0 (rangeID 12, attempt 0)
I230418 10:54:53.491021 926855 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 540  Establishing MuxRangeFeed to node 1
I230418 10:54:53.544703 499144 3@pebble/event.go:717  [n1,s1,pebble] 541  [JOB 7] WAL created 000006
I230418 10:54:53.549257 490673 1@server/server_sql.go:1736  [T1,n1] 542  serving sql connections
--- FAIL: TestRefresh (6.97s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230418 10:54:53.647083 932732 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 543  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.649806 933050 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 544  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.651508 933107 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 545  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.653856 933107 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 546  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.655887 934355 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 547  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.658411 934606 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 548  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.660628 935158 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 549  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.664841 935555 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 550  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.667220 935922 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 551  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.669523 936231 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 552  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.673627 936798 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 553  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.675163 937088 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 554  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.676675 937392 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 555  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.678500 937720 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 556  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.680535 938040 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 557  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.682017 938445 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 558  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.683358 938753 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 559  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.685483 938939 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 560  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.687703 939310 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 561  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.689809 939771 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 562  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.691719 940210 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 563  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.693895 940536 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 564  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.695002 940827 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 565  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.695919 940980 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 566  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.697447 941109 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.698594 941341 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:54:53.699592 941508 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9661641?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9661641?buildTab=artifacts#/) on master @ [1cd507a7c6582fd91bb56123e62bd4fde45c4d22](https://github.com/cockroachdb/cockroach/commits/1cd507a7c6582fd91bb56123e62bd4fde45c4d22):


```
I230418 10:58:53.522715 5507 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.524399 5508 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.525832 5493 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.526927 5494 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.528017 5495 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.529130 5496 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.530256 5497 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.531332 5498 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.532542 5499 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.533640 5500 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.534788 5501 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.535951 5502 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.537257 5503 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 579  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.538893 5509 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 580  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.540574 5510 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 581  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.542233 5511 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 582  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.543768 5512 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 583  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.545372 5513 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 584  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.547015 5514 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.548637 5515 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.550171 5516 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.551844 5517 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.553285 5518 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.554567 5522 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.555737 5523 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.556884 5524 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.558081 5525 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.559178 5526 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.560339 5527 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.561600 5528 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.562801 5529 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.563996 5530 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.565142 5531 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.566396 5532 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.567560 5533 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.568928 5534 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.570311 5535 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:53.571405 5536 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230418 10:58:54.204883 3161 3@pebble/event.go:717  [n1,s1,pebble] 605  [JOB 7] WAL created 000006
I230418 10:58:54.227508 2887 1@server/server_sql.go:1736  [T1,n1] 606  serving sql connections
--- FAIL: TestRefresh (6.63s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9683457?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9683457?buildTab=artifacts#/) on master @ [a4ab6b6243f509e62993b4d81855f23470d9901a](https://github.com/cockroachdb/cockroach/commits/a4ab6b6243f509e62993b4d81855f23470d9901a):


```
I230419 10:44:09.220494 831557 jobs/registry.go:1543  [T1,n1] 409  MIGRATION job 857966670564917249: stepping through state running
I230419 10:44:09.338956 831557 jobs/registry.go:1543  [T1,n1] 410  MIGRATION job 857966670564917249: stepping through state succeeded
I230419 10:44:09.444354 485417 jobs/wait.go:145  [T1,n1] 411  waited for 1 [857966670564917249] queued jobs to complete 299.003742ms
I230419 10:44:09.445997 485417 upgrade/upgrademanager/manager.go:283  [T1,n1] 412  running permanent upgrade for version 1000022.2-90
I230419 10:44:09.566356 485417 jobs/registry.go:613  [T1,n1] 413  active version is 1000023.1-2
I230419 10:44:09.569243 494212 kv/kvserver/replica_rangefeed.go:709  [T1,n1,s1,r22/1:/Table/2{0-1}] 414  RangeFeed closed timestamp is empty
I230419 10:44:09.630557 485417 upgrade/upgrademanager/manager.go:742  [T1,n1] 415  running Upgrade to 1000022.2-90: "create auto config runner job"
I230419 10:44:09.630825 856945 jobs/adopt.go:261  [T1,n1] 416  job 857966672104947713: resuming execution
I230419 10:44:09.708855 859032 jobs/registry.go:1543  [T1,n1] 417  MIGRATION job 857966672104947713: stepping through state running
I230419 10:44:09.777620 859032 jobs/registry.go:613  [T1,n1,job=MIGRATION id=857966672104947713,upgrade=1000022.2-90] 418  active version is 1000023.1-2
I230419 10:44:09.844241 868523 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=857966672104947713,upgrade=1000022.2-90] 419  job 102: resuming execution
I230419 10:44:09.853811 859032 jobs/registry.go:1543  [T1,n1] 420  MIGRATION job 857966672104947713: stepping through state succeeded
I230419 10:44:09.946792 873329 jobs/registry.go:1543  [T1,n1] 421  AUTO CONFIG RUNNER job 102: stepping through state running
I230419 10:44:09.983606 485417 jobs/wait.go:145  [T1,n1] 422  waited for 1 [857966672104947713] queued jobs to complete 352.831725ms
I230419 10:44:09.985637 485417 upgrade/upgrademanager/manager.go:283  [T1,n1] 423  running permanent upgrade for version 1000022.2-98
I230419 10:44:10.012810 873329 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 424  using provider with type acprovider.NoTaskProvider
I230419 10:44:10.012896 873329 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 425  waiting for environment activation...
I230419 10:44:10.013018 873329 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 426  waiting for environment activation...
I230419 10:44:10.127118 485417 jobs/registry.go:613  [T1,n1] 427  active version is 1000023.1-2
I230419 10:44:10.189561 485417 upgrade/upgrademanager/manager.go:742  [T1,n1] 428  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230419 10:44:10.189851 891399 jobs/adopt.go:261  [T1,n1] 429  job 857966673873305601: resuming execution
I230419 10:44:10.282265 893448 jobs/registry.go:1543  [T1,n1] 430  MIGRATION job 857966673873305601: stepping through state running
I230419 10:44:10.374318 899257 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857966673873305601,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 431 ={"Timestamp":1681901050346593056,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:44:10.399971 901699 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857966673873305601,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 432 ={"Timestamp":1681901050376596203,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:44:10.420519 904730 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857966673873305601,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 433 ={"Timestamp":1681901050401985338,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:44:10.443050 906998 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857966673873305601,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 434 ={"Timestamp":1681901050422927084,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:44:10.454296 893448 jobs/registry.go:1543  [T1,n1] 435  MIGRATION job 857966673873305601: stepping through state succeeded
I230419 10:44:10.558265 485417 jobs/wait.go:145  [T1,n1] 436  waited for 1 [857966673873305601] queued jobs to complete 368.475752ms
I230419 10:44:10.558386 485417 server/server_sql.go:1618  [T1,n1] 437  done ensuring all necessary startup migrations have run
I230419 10:44:10.559757 916229 jobs/job_scheduler.go:407  [T1,n1] 438  waiting 3m0s before scheduled jobs daemon start
I230419 10:44:10.563707 916860 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 439  tenant-settings-watcher: established range feed cache
I230419 10:44:10.567839 917532 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 440  tenant-capability-watcher: established range feed cache
I230419 10:44:10.573443 917532 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 441  received results of a full table scan for tenant capabilities
I230419 10:44:10.598469 916325 server/auto_upgrade.go:78  [T1,n1] 442  no need to upgrade, cluster already at the newest version
I230419 10:44:10.606839 916243 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 443  warmed privileges for virtual tables in 47.023083ms
I230419 10:44:10.653841 493949 3@pebble/event.go:717  [n1,s1,pebble] 444  [JOB 7] WAL created 000006
I230419 10:44:10.658411 485417 1@server/server_sql.go:1736  [T1,n1] 445  serving sql connections
--- FAIL: TestRefresh (7.03s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230419 10:44:10.773875 930481 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:44:10.777131 930992 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:44:10.778894 930980 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9683461?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9683461?buildTab=artifacts#/) on master @ [a4ab6b6243f509e62993b4d81855f23470d9901a](https://github.com/cockroachdb/cockroach/commits/a4ab6b6243f509e62993b4d81855f23470d9901a):


```
I230419 10:50:01.894563 5293 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 564  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:50:01.895973 5243 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 565  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:50:01.897276 5244 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 566  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:50:01.898421 5245 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/{Table/61-Max}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230419 10:50:01.904037 5270 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=857967825902731265,upgrade=1000022.2-90] 568  job 102: resuming execution
I230419 10:50:01.908990 4911 jobs/registry.go:1543  [T1,n1] 569  MIGRATION job 857967825902731265: stepping through state succeeded
I230419 10:50:01.958281 5273 jobs/registry.go:1543  [T1,n1] 570  AUTO CONFIG RUNNER job 102: stepping through state running
I230419 10:50:01.986163 2887 jobs/wait.go:145  [T1,n1] 571  waited for 1 [857967825902731265] queued jobs to complete 240.187148ms
I230419 10:50:01.987137 2887 upgrade/upgrademanager/manager.go:283  [T1,n1] 572  running permanent upgrade for version 1000022.2-98
I230419 10:50:02.000029 5273 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 573  using provider with type acprovider.NoTaskProvider
I230419 10:50:02.000219 5273 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 574  waiting for environment activation...
I230419 10:50:02.000415 5273 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 575  waiting for environment activation...
I230419 10:50:02.140742 2887 jobs/registry.go:613  [T1,n1] 576  active version is 1000023.1-2
I230419 10:50:02.182910 2887 upgrade/upgrademanager/manager.go:742  [T1,n1] 577  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230419 10:50:02.183284 5366 jobs/adopt.go:261  [T1,n1] 578  job 857967827312377857: resuming execution
I230419 10:50:02.236509 5370 jobs/registry.go:1543  [T1,n1] 579  MIGRATION job 857967827312377857: stepping through state running
I230419 10:50:02.310019 5417 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857967827312377857,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 580 ={"Timestamp":1681901402283296664,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:50:02.337114 4769 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857967827312377857,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 581 ={"Timestamp":1681901402311894014,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:50:02.352178 5425 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857967827312377857,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 582 ={"Timestamp":1681901402338622347,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:50:02.366675 5309 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857967827312377857,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 583 ={"Timestamp":1681901402354234441,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 10:50:02.373181 5370 jobs/registry.go:1543  [T1,n1] 584  MIGRATION job 857967827312377857: stepping through state succeeded
I230419 10:50:02.448614 2887 jobs/wait.go:145  [T1,n1] 585  waited for 1 [857967827312377857] queued jobs to complete 265.418593ms
I230419 10:50:02.448805 2887 server/server_sql.go:1618  [T1,n1] 586  done ensuring all necessary startup migrations have run
I230419 10:50:02.450068 5476 jobs/job_scheduler.go:407  [T1,n1] 587  waiting 3m0s before scheduled jobs daemon start
I230419 10:50:02.451761 5398 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-boundaries-watcher] 588  Establishing MuxRangeFeed (/Table/5{7-8}@1681901402.450401622,0...; 1 spans)
I230419 10:50:02.452343 5482 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 589  tenant-settings-watcher: established range feed cache
I230419 10:50:02.454225 5464 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-boundaries-watcher] 590  MuxRangeFeed starting for range /Table/5{7-8}@1681901402.450401622,0 (rangeID 58, attempt 0)
I230419 10:50:02.454684 5466 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 591  Establishing MuxRangeFeed to node 1
I230419 10:50:02.454871 5400 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 592  tenant-capability-watcher: established range feed cache
I230419 10:50:02.456352 5400 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 593  received results of a full table scan for tenant capabilities
I230419 10:50:02.472383 5480 server/auto_upgrade.go:78  [T1,n1] 594  no need to upgrade, cluster already at the newest version
I230419 10:50:02.480170 5478 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 595  warmed privileges for virtual tables in 30.0812ms
I230419 10:50:02.502618 5494 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 596  Establishing MuxRangeFeed (/Table/5{0-1}@1681901402.452030689,0...; 1 spans)
I230419 10:50:02.503364 5496 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 597  MuxRangeFeed starting for range /Table/5{0-1}@1681901402.452030689,0 (rangeID 51, attempt 0)
I230419 10:50:02.504000 5497 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 598  Establishing MuxRangeFeed to node 1
I230419 10:50:02.504658 5345 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 599  Establishing MuxRangeFeed (/Table/8/{1-2}@1681901402.454498704,0...; 1 spans)
I230419 10:50:02.505435 5325 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 600  MuxRangeFeed starting for range /Table/8/{1-2}@1681901402.454498704,0 (rangeID 12, attempt 0)
I230419 10:50:02.506032 5326 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 601  Establishing MuxRangeFeed to node 1
I230419 10:50:03.534373 3174 3@pebble/event.go:717  [n1,s1,pebble] 602  [JOB 7] WAL created 000006
I230419 10:50:03.552937 2887 1@server/server_sql.go:1736  [T1,n1] 603  serving sql connections
--- FAIL: TestRefresh (6.63s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9702727?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9702727?buildTab=artifacts#/) on master @ [b767225a6893ea6e81d50618d634dbba92d3b680](https://github.com/cockroachdb/cockroach/commits/b767225a6893ea6e81d50618d634dbba92d3b680):


```
I230420 10:53:54.406409 5749 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.407542 5750 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.408734 5598 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.409900 5599 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.411104 5600 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.412337 5601 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.413458 5778 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.414652 5779 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.415876 5509 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.417010 5510 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.418178 5511 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.419239 5512 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.420377 5513 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 579  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.421629 5514 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 580  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.422774 5515 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 581  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.423837 5516 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 582  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.424921 5517 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 583  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.426101 5518 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 584  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.427249 5519 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.428402 5520 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.429646 5521 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.430744 5794 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.431901 5751 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.433098 5752 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.434322 5753 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.435464 5754 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.436742 5755 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.437942 5756 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.439183 5757 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.440287 5758 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.441490 5759 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.442710 5760 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.443848 5761 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.445067 5795 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.446243 5796 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.447447 5797 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.448646 5798 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:54.449754 5799 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230420 10:53:55.083823 3324 3@pebble/event.go:717  [n1,s1,pebble] 605  [JOB 7] WAL created 000006
I230420 10:53:55.103751 2670 1@server/server_sql.go:1736  [T1,n1] 606  serving sql connections
--- FAIL: TestRefresh (6.47s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9723762?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9723762?buildTab=artifacts#/) on master @ [8124bff03d073f35f4d2b6a2048c7f4417d757d9](https://github.com/cockroachdb/cockroach/commits/8124bff03d073f35f4d2b6a2048c7f4417d757d9):


```
I230421 10:01:50.550225 975231 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-boundaries-watcher] 543  Establishing MuxRangeFeed (/Table/5{7-8}@1682071310.547227210,0...; 1 spans)
I230421 10:01:50.553647 975736 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 544  tenant-settings-watcher: established range feed cache
I230421 10:01:50.554478 975430 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-boundaries-watcher] 545  MuxRangeFeed starting for range /Table/5{7-8}@1682071310.547227210,0 (rangeID 58, attempt 0)
I230421 10:01:50.555056 975834 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 546  Establishing MuxRangeFeed to node 1
I230421 10:01:50.560312 976415 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 547  tenant-capability-watcher: established range feed cache
I230421 10:01:50.569352 976415 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 548  received results of a full table scan for tenant capabilities
I230421 10:01:50.611937 981343 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 549  Establishing MuxRangeFeed (/Table/8/{1-2}@1682071310.559660642,1...; 1 spans)
I230421 10:01:50.615704 981392 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 550  Establishing MuxRangeFeed (/Table/5{0-1}@1682071310.553358549,0...; 1 spans)
I230421 10:01:50.616763 981509 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 551  MuxRangeFeed starting for range /Table/8/{1-2}@1682071310.559660642,1 (rangeID 12, attempt 0)
I230421 10:01:50.619091 981862 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 552  Establishing MuxRangeFeed to node 1
I230421 10:01:50.618344 981805 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 553  MuxRangeFeed starting for range /Table/5{0-1}@1682071310.553358549,0 (rangeID 51, attempt 0)
I230421 10:01:50.622145 982047 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 554  Establishing MuxRangeFeed to node 1
I230421 10:01:50.647338 530304 kv/kvserver/queue.go:614  [T1,n1,s1,r63/1:/{Table/62-Max},raft] 555  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I230421 10:01:50.671743 974983 server/auto_upgrade.go:78  [T1,n1] 556  no need to upgrade, cluster already at the newest version
I230421 10:01:50.681134 974966 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 557  warmed privileges for virtual tables in 133.674413ms
I230421 10:01:50.743422 530420 3@pebble/event.go:717  [n1,s1,pebble] 558  [JOB 7] WAL created 000006
I230421 10:01:50.748128 521810 1@server/server_sql.go:1733  [T1,n1] 559  serving sql connections
--- FAIL: TestRefresh (7.48s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230421 10:01:50.890677 994660 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 560  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.894424 995190 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 561  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.895911 995090 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 562  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.897562 995090 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 563  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.899348 996040 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 564  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.901692 996281 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 565  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.903884 996888 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 566  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.906758 997434 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 567  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.908814 997936 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 568  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.911674 998291 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 569  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.914152 998803 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 570  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.919484 999224 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 571  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.923395 1000065 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 572  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.925557 1000529 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 573  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.927894 1001081 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 574  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.931849 1001563 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 575  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.934354 1002344 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 576  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.937664 1002936 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 577  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.938748 1003124 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 578  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.940004 1002631 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 579  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.942370 1003465 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 580  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.945170 1004047 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 581  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230421 10:01:50.946839 1004204 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 582  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.18s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9741315?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9741315?buildTab=artifacts#/) on master @ [0545913ede29a681dbeef9a6f1211d8b95395c5e](https://github.com/cockroachdb/cockroach/commits/0545913ede29a681dbeef9a6f1211d8b95395c5e):


```
I230422 10:51:07.292126 872008 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 522  waiting for environment activation...
I230422 10:51:07.401483 499668 jobs/registry.go:616  [T1,n1] 523  active version is 1000023.1-2
I230422 10:51:07.462062 499668 upgrade/upgrademanager/manager.go:742  [T1,n1] 524  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230422 10:51:07.462335 890247 jobs/adopt.go:261  [T1,n1] 525  job 858817387777064961: resuming execution
I230422 10:51:07.539163 892309 jobs/registry.go:1584  [T1,n1] 526  MIGRATION job 858817387777064961: stepping through state running
I230422 10:51:07.615095 898051 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858817387777064961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 527 ={"Timestamp":1682160667598346799,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230422 10:51:07.637235 899389 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858817387777064961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 528 ={"Timestamp":1682160667616755611,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230422 10:51:07.658289 901744 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858817387777064961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 529 ={"Timestamp":1682160667639165556,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230422 10:51:07.677432 904040 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858817387777064961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 530 ={"Timestamp":1682160667659973697,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230422 10:51:07.688344 892309 jobs/registry.go:1584  [T1,n1] 531  MIGRATION job 858817387777064961: stepping through state succeeded
I230422 10:51:07.785948 499668 jobs/wait.go:145  [T1,n1] 532  waited for 1 [858817387777064961] queued jobs to complete 323.661633ms
I230422 10:51:07.787642 499668 upgrade/upgrademanager/manager.go:283  [T1,n1] 533  running permanent upgrade for version 1000022.2-102
I230422 10:51:07.905634 499668 jobs/registry.go:616  [T1,n1] 534  active version is 1000023.1-2
I230422 10:51:07.961572 499668 upgrade/upgrademanager/manager.go:742  [T1,n1] 535  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230422 10:51:07.961873 923802 jobs/adopt.go:261  [T1,n1] 536  job 858817389486866433: resuming execution
I230422 10:51:08.039933 925929 jobs/registry.go:1584  [T1,n1] 537  MIGRATION job 858817389486866433: stepping through state running
I230422 10:51:08.164787 925929 jobs/registry.go:1584  [T1,n1] 538  MIGRATION job 858817389486866433: stepping through state succeeded
I230422 10:51:08.260629 499668 jobs/wait.go:145  [T1,n1] 539  waited for 1 [858817389486866433] queued jobs to complete 298.823536ms
I230422 10:51:08.260752 499668 server/server_sql.go:1615  [T1,n1] 540  done ensuring all necessary startup migrations have run
I230422 10:51:08.262997 940584 jobs/job_scheduler.go:407  [T1,n1] 541  waiting 2m0s before scheduled jobs daemon start
I230422 10:51:08.265882 940999 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-boundaries-watcher] 542  Establishing MuxRangeFeed (/Table/5{7-8}@1682160668.263407679,1...; 1 spans)
I230422 10:51:08.268041 941365 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 543  tenant-settings-watcher: established range feed cache
I230422 10:51:08.269294 941060 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-boundaries-watcher] 544  MuxRangeFeed starting for range /Table/5{7-8}@1682160668.263407679,1 (rangeID 58, attempt 0)
I230422 10:51:08.269476 941622 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 545  Establishing MuxRangeFeed to node 1
I230422 10:51:08.272072 942050 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 546  tenant-capability-watcher: established range feed cache
I230422 10:51:08.276102 942050 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 547  received results of a full table scan for tenant capabilities
I230422 10:51:08.298565 940650 server/auto_upgrade.go:78  [T1,n1] 548  no need to upgrade, cluster already at the newest version
I230422 10:51:08.306627 940626 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 549  warmed privileges for virtual tables in 43.509711ms
I230422 10:51:08.316402 947465 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 550  Establishing MuxRangeFeed (/Table/5{0-1}@1682160668.267765765,0...; 1 spans)
I230422 10:51:08.316690 947885 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 551  MuxRangeFeed starting for range /Table/5{0-1}@1682160668.267765765,0 (rangeID 51, attempt 0)
I230422 10:51:08.316913 948039 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 552  Establishing MuxRangeFeed to node 1
I230422 10:51:08.325278 948198 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 553  Establishing MuxRangeFeed (/Table/8/{1-2}@1682160668.271797843,0...; 1 spans)
I230422 10:51:08.325536 948207 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 554  MuxRangeFeed starting for range /Table/8/{1-2}@1682160668.271797843,0 (rangeID 12, attempt 0)
I230422 10:51:08.325708 948457 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 555  Establishing MuxRangeFeed to node 1
I230422 10:51:08.364281 508209 3@pebble/event.go:717  [n1,s1,pebble] 556  [JOB 7] WAL created 000006
I230422 10:51:08.368146 499668 1@server/server_sql.go:1733  [T1,n1] 557  serving sql connections
--- FAIL: TestRefresh (7.19s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230422 10:51:08.490264 948485 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 558  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230422 10:51:08.497658 955681 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 559  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230422 10:51:08.504757 955599 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 560  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230422 10:51:08.512808 955599 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 561  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9753673?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9753673?buildTab=artifacts#/) on master @ [1f3419e178bdba544f74d9c9e14a4682efd18028](https://github.com/cockroachdb/cockroach/commits/1f3419e178bdba544f74d9c9e14a4682efd18028):


```
I230423 11:47:40.362618 520896 upgrade/upgrademanager/manager.go:283  [T1,n1] 485  running permanent upgrade for version 1000022.2-90
I230423 11:47:40.475720 520896 jobs/registry.go:616  [T1,n1] 486  active version is 1000023.1-2
I230423 11:47:40.537099 520896 upgrade/upgrademanager/manager.go:742  [T1,n1] 487  running Upgrade to 1000022.2-90: "create auto config runner job"
I230423 11:47:40.537377 899599 jobs/adopt.go:261  [T1,n1] 488  job 859111621796560897: resuming execution
I230423 11:47:40.621736 902389 jobs/registry.go:1584  [T1,n1] 489  MIGRATION job 859111621796560897: stepping through state running
I230423 11:47:40.685311 902389 jobs/registry.go:616  [T1,n1,job=MIGRATION id=859111621796560897,upgrade=1000022.2-90] 490  active version is 1000023.1-2
I230423 11:47:40.743560 912115 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=859111621796560897,upgrade=1000022.2-90] 491  job 102: resuming execution
I230423 11:47:40.752558 902389 jobs/registry.go:1584  [T1,n1] 492  MIGRATION job 859111621796560897: stepping through state succeeded
I230423 11:47:40.839129 916036 jobs/registry.go:1584  [T1,n1] 493  AUTO CONFIG RUNNER job 102: stepping through state running
I230423 11:47:40.874659 520896 jobs/wait.go:145  [T1,n1] 494  waited for 1 [859111621796560897] queued jobs to complete 337.335171ms
I230423 11:47:40.876396 520896 upgrade/upgrademanager/manager.go:283  [T1,n1] 495  running permanent upgrade for version 1000022.2-98
I230423 11:47:40.901285 916036 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 496  using provider with type acprovider.NoTaskProvider
I230423 11:47:40.901400 916036 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 497  waiting for environment activation...
I230423 11:47:40.901494 916036 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 498  waiting for environment activation...
I230423 11:47:41.022140 520896 jobs/registry.go:616  [T1,n1] 499  active version is 1000023.1-2
I230423 11:47:41.092705 520896 upgrade/upgrademanager/manager.go:742  [T1,n1] 500  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230423 11:47:41.092979 935888 jobs/adopt.go:261  [T1,n1] 501  job 859111623480016897: resuming execution
I230423 11:47:41.170150 938261 jobs/registry.go:1584  [T1,n1] 502  MIGRATION job 859111623480016897: stepping through state running
I230423 11:47:41.251332 943891 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=859111623480016897,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 503 ={"Timestamp":1682250461232915398,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230423 11:47:41.275623 945186 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=859111623480016897,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 504 ={"Timestamp":1682250461253341057,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230423 11:47:41.297053 947425 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=859111623480016897,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 505 ={"Timestamp":1682250461277316211,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230423 11:47:41.320024 949676 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=859111623480016897,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 506 ={"Timestamp":1682250461299466163,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230423 11:47:41.331063 938261 jobs/registry.go:1584  [T1,n1] 507  MIGRATION job 859111623480016897: stepping through state succeeded
I230423 11:47:41.432575 520896 jobs/wait.go:145  [T1,n1] 508  waited for 1 [859111623480016897] queued jobs to complete 339.639022ms
I230423 11:47:41.434196 520896 upgrade/upgrademanager/manager.go:283  [T1,n1] 509  running permanent upgrade for version 1000022.2-102
I230423 11:47:41.541019 520896 jobs/registry.go:616  [T1,n1] 510  active version is 1000023.1-2
I230423 11:47:41.627382 520896 upgrade/upgrademanager/manager.go:742  [T1,n1] 511  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230423 11:47:41.627609 971384 jobs/adopt.go:261  [T1,n1] 512  job 859111625307815937: resuming execution
I230423 11:47:41.711830 973455 jobs/registry.go:1584  [T1,n1] 513  MIGRATION job 859111625307815937: stepping through state running
I230423 11:47:41.842472 973455 jobs/registry.go:1584  [T1,n1] 514  MIGRATION job 859111625307815937: stepping through state succeeded
I230423 11:47:41.951993 520896 jobs/wait.go:145  [T1,n1] 515  waited for 1 [859111625307815937] queued jobs to complete 324.423152ms
I230423 11:47:41.952141 520896 server/server_sql.go:1615  [T1,n1] 516  done ensuring all necessary startup migrations have run
I230423 11:47:41.954127 988374 jobs/job_scheduler.go:407  [T1,n1] 517  waiting 2m0s before scheduled jobs daemon start
I230423 11:47:41.958565 989029 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 518  tenant-settings-watcher: established range feed cache
I230423 11:47:41.962371 989679 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 519  tenant-capability-watcher: established range feed cache
I230423 11:47:41.967050 989679 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 520  received results of a full table scan for tenant capabilities
I230423 11:47:41.999500 988424 server/auto_upgrade.go:78  [T1,n1] 521  no need to upgrade, cluster already at the newest version
I230423 11:47:42.008705 988384 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 522  warmed privileges for virtual tables in 54.529656ms
I230423 11:47:42.061662 529477 3@pebble/event.go:717  [n1,s1,pebble] 523  [JOB 7] WAL created 000006
I230423 11:47:42.066656 520896 1@server/server_sql.go:1733  [T1,n1] 524  serving sql connections
--- FAIL: TestRefresh (7.81s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9753678?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9753678?buildTab=artifacts#/) on master @ [1f3419e178bdba544f74d9c9e14a4682efd18028](https://github.com/cockroachdb/cockroach/commits/1f3419e178bdba544f74d9c9e14a4682efd18028):


```
I230423 11:50:51.228101 6072 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 584  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.229202 6073 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.230397 6074 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.231600 6075 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.232716 6076 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.233822 6077 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.234952 6078 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.236110 6079 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.237269 6080 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.238412 6081 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.239651 6082 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.240749 6083 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.241847 6084 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.243037 6085 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.244203 6086 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.245371 6087 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.246494 6088 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.247608 6089 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.248713 6090 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.249896 6091 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.250981 6092 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.252117 6093 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 605  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.253240 6094 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 606  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.254411 6095 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 607  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.255533 6096 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 608  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.256661 6097 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 609  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.257855 6098 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 610  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.259063 6099 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 611  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.260351 6100 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 612  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.261605 6101 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 613  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.262749 6102 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 614  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.263973 6103 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 615  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.265140 6104 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 616  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.266329 6105 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 617  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.267587 6106 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 618  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.268754 6107 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 619  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.269936 6108 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 620  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.271254 6109 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 621  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230423 11:50:51.969284 3478 3@pebble/event.go:717  [n1,s1,pebble] 622  [JOB 7] WAL created 000006
I230423 11:50:51.996650 2806 1@server/server_sql.go:1733  [T1,n1] 623  serving sql connections
--- FAIL: TestRefresh (7.51s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:627
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9763732?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9763732?buildTab=artifacts#/) on master @ [3c23346a6d1c91dfe8ed1c5285966f9b9e487601](https://github.com/cockroachdb/cockroach/commits/3c23346a6d1c91dfe8ed1c5285966f9b9e487601):


```
I230424 10:21:15.837124 6019 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 585  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.838615 6020 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 586  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.839905 6021 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 587  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.841211 6022 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 588  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.843810 5817 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 589  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.845701 5818 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 590  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.848970 5696 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 591  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.850569 5697 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 592  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.852198 6050 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 593  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.853646 5800 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 594  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.854788 5801 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 595  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.855916 5802 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 596  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.857145 5803 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 597  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.858285 5804 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 598  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.859384 5805 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 599  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.860531 5806 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 600  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.861644 5807 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 601  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.862732 5808 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 602  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.863868 5809 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 603  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.865053 6066 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 604  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.866163 6067 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 605  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.867260 6068 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 606  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.868450 6069 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 607  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.869590 6070 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 608  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.870764 6071 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 609  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.871854 6072 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 610  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.873021 6073 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 611  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.874278 6074 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 612  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.875423 6075 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 613  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.876571 6076 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 614  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.877737 6077 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 615  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.878877 6078 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 616  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.879989 6079 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 617  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.881257 6080 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 618  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.882396 6081 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 619  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.883540 6082 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 620  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.884784 6083 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 621  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:15.885938 6084 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 622  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230424 10:21:16.708771 3416 3@pebble/event.go:717  [n1,s1,pebble] 623  [JOB 7] WAL created 000006
I230424 10:21:16.729518 2586 1@server/server_sql.go:1733  [T1,n1] 624  serving sql connections
--- FAIL: TestRefresh (7.48s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9785107?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9785107?buildTab=artifacts#/) on master @ [47ac4bf28d8cdfef32bd4ffa4ce10da45bfdfbd8](https://github.com/cockroachdb/cockroach/commits/47ac4bf28d8cdfef32bd4ffa4ce10da45bfdfbd8):


```
I230425 10:59:14.944943 6102 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.946423 6103 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.947692 6104 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.948965 6105 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.950212 6106 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.951481 6107 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.952636 6108 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.953870 6109 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.955044 6110 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.956511 6111 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.957895 6112 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.959276 6113 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.960437 6114 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.961529 6115 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.962697 6116 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.963815 6117 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.964848 6118 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.965945 6119 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.967159 6120 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.968266 6121 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.969433 6122 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.970787 6123 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.972385 6124 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 509  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.973856 6125 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.975549 5815 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 511  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.977151 5816 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.978510 5817 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 513  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.979768 5818 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 514  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.981070 5819 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 515  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.982330 5820 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 516  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.983665 5821 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 517  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.984868 5822 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 518  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.986106 5823 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 519  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.987322 5824 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 520  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.988481 5825 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 521  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.989700 6130 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 522  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.990845 6131 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 523  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:14.992027 6132 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 524  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230425 10:59:15.670043 3553 3@pebble/event.go:717  [n1,s1,pebble] 525  [JOB 7] WAL created 000006
I230425 10:59:15.700772 3177 1@server/server_sql.go:1733  [T1,n1] 526  serving sql connections
--- FAIL: TestRefresh (7.56s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:627
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9806925?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9806925?buildTab=artifacts#/) on master @ [9fb03ee9b365d656cc26728cc87f682a385d511b](https://github.com/cockroachdb/cockroach/commits/9fb03ee9b365d656cc26728cc87f682a385d511b):


```
I230426 11:16:42.988142 6087 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.989612 6088 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.991033 6089 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.992286 6090 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.993723 6091 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.995026 6092 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.996431 6093 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.997714 6094 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:42.999156 6095 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.000453 6143 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.001831 6144 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.002968 6145 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.004109 6162 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.005663 6163 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.006943 6164 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.008396 6165 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.009765 6166 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.011058 6167 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.012475 6168 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.013691 6169 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.015050 6170 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.016399 6171 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.017785 6172 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.019028 6173 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 509  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.020120 6174 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.021366 6175 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 511  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.022796 6176 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.023878 6177 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 513  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.025216 6178 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 514  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.026503 6179 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 515  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.027767 6180 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 516  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.029229 6181 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 517  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.030431 6182 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 518  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.031691 6183 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 519  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.033129 6184 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 520  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.034388 6185 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 521  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.035787 6186 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 522  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.037051 6187 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 523  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230426 11:16:43.878608 3559 3@pebble/event.go:717  [n1,s1,pebble] 524  [JOB 7] WAL created 000006
I230426 11:16:43.900060 3160 1@server/server_sql.go:1733  [T1,n1] 525  serving sql connections
--- FAIL: TestRefresh (7.59s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9829169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9829169?buildTab=artifacts#/) on master @ [5b44c4fd6d97dc3a75e786fed24f47ad8bcf9039](https://github.com/cockroachdb/cockroach/commits/5b44c4fd6d97dc3a75e786fed24f47ad8bcf9039):


```
--- FAIL: TestRefresh (7.35s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230427 10:55:11.067724 934312 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 397  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.069659 934641 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 398  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.071693 934576 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 399  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.072889 934576 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 400  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.075158 935714 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 401  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.076934 936088 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 402  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.078897 936365 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 403  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.080608 936922 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 404  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.082265 937356 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 405  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.083971 937686 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 406  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.085396 938023 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 407  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.087149 938406 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 408  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.089450 938526 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 409  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.091626 939267 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 410  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.093266 939684 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 411  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.095179 939999 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 412  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.096695 940442 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 413  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.098601 940852 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 414  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.099680 941167 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 415  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.101545 941570 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 416  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.103098 941910 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 417  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.104603 942237 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 418  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.105733 942544 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 419  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.106756 942774 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 420  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.107764 942901 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 421  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.108800 942993 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 422  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.109827 943147 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 423  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.111078 943384 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 424  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.112229 943520 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 425  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.113207 943658 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 426  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.114095 943758 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 427  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.115293 943940 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 428  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.116651 944138 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 429  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.117820 944356 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 430  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.118944 944318 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 431  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.120087 944696 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 432  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.120999 944855 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 433  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.121867 944974 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 434  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.123375 945150 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:55:11.124537 945442 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9829174?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9829174?buildTab=artifacts#/) on master @ [5b44c4fd6d97dc3a75e786fed24f47ad8bcf9039](https://github.com/cockroachdb/cockroach/commits/5b44c4fd6d97dc3a75e786fed24f47ad8bcf9039):


```
I230427 10:59:05.648562 5766 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.649754 5767 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.650921 5768 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.652089 5769 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.653271 5770 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.654411 5771 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.655645 5772 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.656848 5773 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.657967 5774 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.659145 5775 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.660335 5776 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.661507 5777 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.662688 5778 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.663867 5779 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.665250 5780 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.666456 5781 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.667726 5782 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.668962 5783 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.670251 5784 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.671465 5785 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.672661 5786 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.673818 5787 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.674961 5788 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 509  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.676142 5789 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.677362 5790 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 511  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.678479 5791 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.679616 5792 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 513  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.681014 5793 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 514  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.682397 5794 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 515  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.683536 5795 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 516  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.684592 5796 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 517  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.685716 5797 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 518  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.686890 5798 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 519  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.688166 5799 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 520  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.689719 5800 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 521  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.691125 5801 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 522  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.692275 5802 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 523  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:05.693496 5803 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 524  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230427 10:59:06.371183 3329 3@pebble/event.go:717  [n1,s1,pebble] 525  [JOB 7] WAL created 000006
I230427 10:59:06.397146 2590 1@server/server_sql.go:1733  [T1,n1] 526  serving sql connections
--- FAIL: TestRefresh (7.59s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:627
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9852111?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9852111?buildTab=artifacts#/) on master @ [3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f](https://github.com/cockroachdb/cockroach/commits/3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f):


```
I230428 11:15:47.825971 849848 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 370  waiting for environment activation...
I230428 11:15:47.931949 482116 jobs/registry.go:617  [T1,n1] 371  active version is 1000023.1-6
I230428 11:15:47.992669 482116 upgrade/upgrademanager/manager.go:742  [T1,n1] 372  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230428 11:15:47.993009 867831 jobs/adopt.go:261  [T1,n1] 373  job 860520932301996033: resuming execution
I230428 11:15:48.069180 869921 jobs/registry.go:1586  [T1,n1] 374  MIGRATION job 860520932301996033: stepping through state running
I230428 11:15:48.151126 875537 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860520932301996033,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 375 ={"Timestamp":1682680548132963567,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230428 11:15:48.173655 876878 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860520932301996033,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 376 ={"Timestamp":1682680548152874831,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230428 11:15:48.193971 879387 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860520932301996033,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 377 ={"Timestamp":1682680548175597255,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230428 11:15:48.214384 881715 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860520932301996033,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 378 ={"Timestamp":1682680548196119482,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230428 11:15:48.224513 869921 jobs/registry.go:1586  [T1,n1] 379  MIGRATION job 860520932301996033: stepping through state succeeded
I230428 11:15:48.328103 482116 jobs/wait.go:145  [T1,n1] 380  waited for 1 [860520932301996033] queued jobs to complete 335.151853ms
I230428 11:15:48.329931 482116 upgrade/upgrademanager/manager.go:283  [T1,n1] 381  running permanent upgrade for version 1000022.2-102
I230428 11:15:48.445335 482116 jobs/registry.go:617  [T1,n1] 382  active version is 1000023.1-6
I230428 11:15:48.501509 482116 upgrade/upgrademanager/manager.go:742  [T1,n1] 383  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230428 11:15:48.501818 900933 jobs/adopt.go:261  [T1,n1] 384  job 860520934047842305: resuming execution
I230428 11:15:48.569895 902978 jobs/registry.go:1586  [T1,n1] 385  MIGRATION job 860520934047842305: stepping through state running
I230428 11:15:48.687959 902978 jobs/registry.go:1586  [T1,n1] 386  MIGRATION job 860520934047842305: stepping through state succeeded
I230428 11:15:48.784183 482116 jobs/wait.go:145  [T1,n1] 387  waited for 1 [860520934047842305] queued jobs to complete 282.434016ms
I230428 11:15:48.784333 482116 server/server_sql.go:1615  [T1,n1] 388  done ensuring all necessary startup migrations have run
I230428 11:15:48.786894 917504 jobs/job_scheduler.go:407  [T1,n1] 389  waiting 2m0s before scheduled jobs daemon start
I230428 11:15:48.791833 918204 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 390  tenant-settings-watcher: established range feed cache
I230428 11:15:48.800559 918983 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 391  tenant-capability-watcher: established range feed cache
I230428 11:15:48.811360 918983 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 392  received results of a full table scan for tenant capabilities
I230428 11:15:48.840182 917593 server/auto_upgrade.go:78  [T1,n1] 393  no need to upgrade, cluster already at the newest version
I230428 11:15:48.847288 917516 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 394  warmed privileges for virtual tables in 60.262224ms
I230428 11:15:48.912516 490819 3@pebble/event.go:717  [n1,s1,pebble] 395  [JOB 7] WAL created 000006
I230428 11:15:48.918056 482116 1@server/server_sql.go:1733  [T1,n1] 396  serving sql connections
--- FAIL: TestRefresh (7.17s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230428 11:15:49.030963 931774 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 397  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.033283 932080 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 398  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.034945 932110 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 399  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.036640 932110 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 400  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.038591 933229 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 401  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.040664 933654 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 402  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.042407 934165 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 403  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.043983 934334 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 404  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.045279 934827 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 405  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.046983 935116 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 406  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.048741 935454 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 407  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.050865 935928 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 408  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:15:49.052470 936395 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 409  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9852114?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9852114?buildTab=artifacts#/) on master @ [3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f](https://github.com/cockroachdb/cockroach/commits/3b1bc872bb4bf8e5eb2ad8790ba8ec781167dd1f):


```
I230428 11:37:25.042613 5683 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.043595 5684 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.044497 5685 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.045487 5686 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.046512 5687 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.047473 5688 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.048612 5689 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.049752 5690 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.050875 5404 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.051929 5405 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.052925 5406 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.053870 5407 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.054815 5408 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.055772 5409 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.056699 5698 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.057642 5699 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.058599 5700 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.059648 5701 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.060634 5702 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.061686 5703 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.062562 5704 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.063483 5705 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.064450 5706 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.065334 5707 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.066258 5708 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.067195 5709 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 509  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.068159 5710 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.069017 5711 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 511  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.069971 5712 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.070959 5713 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 513  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.071945 5714 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 514  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.072980 5715 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 515  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.073986 5716 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 516  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.074935 5717 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 517  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.075870 5718 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 518  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.076789 5719 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 519  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.077710 5720 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 520  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.078733 5721 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 521  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230428 11:37:25.589103 3219 3@pebble/event.go:717  [n1,s1,pebble] 522  [JOB 7] WAL created 000006
I230428 11:37:25.608743 1325 1@server/server_sql.go:1733  [T1,n1] 523  serving sql connections
--- FAIL: TestRefresh (6.64s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9871709?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9871709?buildTab=artifacts#/) on master @ [4619767bb0b75ac85ddcb76e33c73211e369afed](https://github.com/cockroachdb/cockroach/commits/4619767bb0b75ac85ddcb76e33c73211e369afed):


```
I230429 11:20:32.799587 884240 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860804980573995009,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 436 ={"Timestamp":1682767232782007999,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230429 11:20:32.820920 885566 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860804980573995009,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 437 ={"Timestamp":1682767232800997582,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230429 11:20:32.839555 887872 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860804980573995009,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 438 ={"Timestamp":1682767232822852295,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230429 11:20:32.858491 890221 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=860804980573995009,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 439 ={"Timestamp":1682767232841632122,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230429 11:20:32.868508 878416 jobs/registry.go:1586  [T1,n1] 440  MIGRATION job 860804980573995009: stepping through state succeeded
I230429 11:20:32.963893 486164 jobs/wait.go:145  [T1,n1] 441  waited for 1 [860804980573995009] queued jobs to complete 319.061288ms
I230429 11:20:32.965249 486164 upgrade/upgrademanager/manager.go:283  [T1,n1] 442  running permanent upgrade for version 1000022.2-102
I230429 11:20:33.078309 486164 jobs/registry.go:617  [T1,n1] 443  active version is 1000023.1-6
I230429 11:20:33.144164 486164 upgrade/upgrademanager/manager.go:742  [T1,n1] 444  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230429 11:20:33.144456 908256 jobs/adopt.go:261  [T1,n1] 445  job 860804982260793345: resuming execution
I230429 11:20:33.226930 910553 jobs/registry.go:1586  [T1,n1] 446  MIGRATION job 860804982260793345: stepping through state running
I230429 11:20:33.336578 910553 jobs/registry.go:1586  [T1,n1] 447  MIGRATION job 860804982260793345: stepping through state succeeded
I230429 11:20:33.430310 486164 jobs/wait.go:145  [T1,n1] 448  waited for 1 [860804982260793345] queued jobs to complete 285.930101ms
I230429 11:20:33.430398 486164 server/server_sql.go:1615  [T1,n1] 449  done ensuring all necessary startup migrations have run
I230429 11:20:33.432097 926220 jobs/job_scheduler.go:407  [T1,n1] 450  waiting 2m0s before scheduled jobs daemon start
I230429 11:20:33.436578 926873 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 451  tenant-settings-watcher: established range feed cache
I230429 11:20:33.440108 927526 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 452  tenant-capability-watcher: established range feed cache
I230429 11:20:33.443814 927526 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 453  received results of a full table scan for tenant capabilities
I230429 11:20:33.465844 926239 server/auto_upgrade.go:78  [T1,n1] 454  no need to upgrade, cluster already at the newest version
I230429 11:20:33.475630 915578 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 455  warmed privileges for virtual tables in 43.344593ms
I230429 11:20:33.543442 494749 3@pebble/event.go:717  [n1,s1,pebble] 456  [JOB 7] WAL created 000006
I230429 11:20:33.548627 486164 1@server/server_sql.go:1733  [T1,n1] 457  serving sql connections
--- FAIL: TestRefresh (6.94s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230429 11:20:33.670530 939440 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.672142 940137 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.674034 940224 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.675590 940224 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.677659 941320 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.679261 941731 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.680882 942078 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.682301 942416 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.683899 942718 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.685369 942986 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.686760 943278 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.688860 943683 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.690496 944097 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.693315 944565 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.694978 945127 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.696481 945427 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.697951 945750 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230429 11:20:33.699977 945964 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.14s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9883252?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9883252?buildTab=artifacts#/) on master @ [7d02e061759599a2b054ee06c71524443f77d20a](https://github.com/cockroachdb/cockroach/commits/7d02e061759599a2b054ee06c71524443f77d20a):


```
I230430 11:09:44.825610 836283 jobs/registry.go:1586  [T1,n1] 429  MIGRATION job 861085972641873921: stepping through state succeeded
I230430 11:09:44.914940 850031 jobs/registry.go:1586  [T1,n1] 430  AUTO CONFIG RUNNER job 102: stepping through state running
I230430 11:09:44.940402 492392 jobs/wait.go:145  [T1,n1] 431  waited for 1 [861085972641873921] queued jobs to complete 333.503807ms
I230430 11:09:44.942253 492392 upgrade/upgrademanager/manager.go:283  [T1,n1] 432  running permanent upgrade for version 1000022.2-98
I230430 11:09:44.975728 850031 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 433  using provider with type acprovider.NoTaskProvider
I230430 11:09:44.975823 850031 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 434  waiting for environment activation...
I230430 11:09:44.975923 850031 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 435  waiting for environment activation...
I230430 11:09:45.094381 492392 jobs/registry.go:617  [T1,n1] 436  active version is 1000023.1-6
I230430 11:09:45.159924 492392 upgrade/upgrademanager/manager.go:742  [T1,n1] 437  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230430 11:09:45.160280 867324 jobs/adopt.go:261  [T1,n1] 438  job 861085974339158017: resuming execution
I230430 11:09:45.257050 869832 jobs/registry.go:1586  [T1,n1] 439  MIGRATION job 861085974339158017: stepping through state running
I230430 11:09:45.351746 875734 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861085974339158017,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 440 ={"Timestamp":1682852985333846276,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230430 11:09:45.375452 877046 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861085974339158017,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 441 ={"Timestamp":1682852985353533213,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230430 11:09:45.410149 879337 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861085974339158017,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 442 ={"Timestamp":1682852985377109632,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230430 11:09:45.430091 881669 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861085974339158017,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 443 ={"Timestamp":1682852985411680230,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230430 11:09:45.440494 869832 jobs/registry.go:1586  [T1,n1] 444  MIGRATION job 861085974339158017: stepping through state succeeded
I230430 11:09:45.552454 492392 jobs/wait.go:145  [T1,n1] 445  waited for 1 [861085974339158017] queued jobs to complete 392.228646ms
I230430 11:09:45.554393 492392 upgrade/upgrademanager/manager.go:283  [T1,n1] 446  running permanent upgrade for version 1000022.2-102
I230430 11:09:45.694070 492392 jobs/registry.go:617  [T1,n1] 447  active version is 1000023.1-6
I230430 11:09:45.761557 492392 upgrade/upgrademanager/manager.go:742  [T1,n1] 448  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230430 11:09:45.761973 900709 jobs/adopt.go:261  [T1,n1] 449  job 861085976345018369: resuming execution
I230430 11:09:45.845947 902910 jobs/registry.go:1586  [T1,n1] 450  MIGRATION job 861085976345018369: stepping through state running
I230430 11:09:45.965708 902910 jobs/registry.go:1586  [T1,n1] 451  MIGRATION job 861085976345018369: stepping through state succeeded
I230430 11:09:46.069475 492392 jobs/wait.go:145  [T1,n1] 452  waited for 1 [861085976345018369] queued jobs to complete 307.56205ms
I230430 11:09:46.069630 492392 server/server_sql.go:1615  [T1,n1] 453  done ensuring all necessary startup migrations have run
I230430 11:09:46.072030 917613 jobs/job_scheduler.go:407  [T1,n1] 454  waiting 2m0s before scheduled jobs daemon start
I230430 11:09:46.077021 918406 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 455  tenant-settings-watcher: established range feed cache
I230430 11:09:46.080426 919045 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 456  tenant-capability-watcher: established range feed cache
I230430 11:09:46.084746 919045 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 457  received results of a full table scan for tenant capabilities
I230430 11:09:46.106581 917676 server/auto_upgrade.go:78  [T1,n1] 458  no need to upgrade, cluster already at the newest version
I230430 11:09:46.117622 913976 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 459  warmed privileges for virtual tables in 45.431278ms
I230430 11:09:46.189273 500985 3@pebble/event.go:717  [n1,s1,pebble] 460  [JOB 7] WAL created 000006
I230430 11:09:46.193912 492392 1@server/server_sql.go:1733  [T1,n1] 461  serving sql connections
--- FAIL: TestRefresh (7.34s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230430 11:09:46.311743 931857 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.313452 932314 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.315237 932293 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.317160 932293 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.318871 933286 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.320733 933631 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:09:46.322236 933933 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9883257?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9883257?buildTab=artifacts#/) on master @ [7d02e061759599a2b054ee06c71524443f77d20a](https://github.com/cockroachdb/cockroach/commits/7d02e061759599a2b054ee06c71524443f77d20a):


```
I230430 11:10:55.122927 5779 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.124153 5780 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.125209 5781 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.126305 5782 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.127504 5783 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.128668 5784 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.129755 5785 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.130881 5786 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.131939 5787 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.133027 5788 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 498  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.134122 5789 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.135202 5790 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.136258 5791 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.137381 5792 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 502  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.138482 5793 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 503  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.139662 5794 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 504  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.140868 5795 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 505  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.141932 5796 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 506  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.142991 5797 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 507  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.144084 5798 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.145205 5799 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 509  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.146292 5800 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.147473 5801 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 511  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.148665 5802 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.149733 5803 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 513  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.150813 5804 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 514  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.151928 5805 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 515  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.153005 5806 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 516  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.154098 5807 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 517  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.155211 5808 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 518  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.156306 5809 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 519  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.157415 5810 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 520  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.158540 5811 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 521  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.159674 5812 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 522  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.160846 5813 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 523  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.161995 5814 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 524  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.163053 5815 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 525  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.164330 5749 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 526  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230430 11:10:55.818325 3300 3@pebble/event.go:717  [n1,s1,pebble] 527  [JOB 7] WAL created 000006
I230430 11:10:55.843525 2843 1@server/server_sql.go:1733  [T1,n1] 528  serving sql connections
--- FAIL: TestRefresh (7.71s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9894268?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9894268?buildTab=artifacts#/) on master @ [ce40ce73c965fdcdb3bda956ba50ad5aca9efefc](https://github.com/cockroachdb/cockroach/commits/ce40ce73c965fdcdb3bda956ba50ad5aca9efefc):


```
I230501 11:35:23.222628 868031 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861374129357193217,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 442 ={"Timestamp":1682940923203777626,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230501 11:35:23.245072 869474 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861374129357193217,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 443 ={"Timestamp":1682940923224497827,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230501 11:35:23.265239 871736 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861374129357193217,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 444 ={"Timestamp":1682940923246532311,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230501 11:35:23.285514 874003 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861374129357193217,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 445 ={"Timestamp":1682940923267382177,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230501 11:35:23.295217 862271 jobs/registry.go:1586  [T1,n1] 446  MIGRATION job 861374129357193217: stepping through state succeeded
I230501 11:35:23.392663 485049 jobs/wait.go:145  [T1,n1] 447  waited for 1 [861374129357193217] queued jobs to complete 323.515376ms
I230501 11:35:23.394149 485049 upgrade/upgrademanager/manager.go:283  [T1,n1] 448  running permanent upgrade for version 1000022.2-102
I230501 11:35:23.512876 485049 jobs/registry.go:617  [T1,n1] 449  active version is 1000023.1-6
I230501 11:35:23.581034 485049 upgrade/upgrademanager/manager.go:742  [T1,n1] 450  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230501 11:35:23.581397 891517 jobs/adopt.go:261  [T1,n1] 451  job 861374131058147329: resuming execution
I230501 11:35:23.660111 893992 jobs/registry.go:1586  [T1,n1] 452  MIGRATION job 861374131058147329: stepping through state running
I230501 11:35:23.779124 893992 jobs/registry.go:1586  [T1,n1] 453  MIGRATION job 861374131058147329: stepping through state succeeded
I230501 11:35:23.881999 485049 jobs/wait.go:145  [T1,n1] 454  waited for 1 [861374131058147329] queued jobs to complete 300.655339ms
I230501 11:35:23.882146 485049 server/server_sql.go:1615  [T1,n1] 455  done ensuring all necessary startup migrations have run
I230501 11:35:23.884622 909538 jobs/job_scheduler.go:407  [T1,n1] 456  waiting 3m0s before scheduled jobs daemon start
I230501 11:35:23.889615 910269 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 457  tenant-settings-watcher: established range feed cache
I230501 11:35:23.893113 910912 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 458  tenant-capability-watcher: established range feed cache
I230501 11:35:23.897593 910912 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 459  received results of a full table scan for tenant capabilities
I230501 11:35:23.915799 909606 server/auto_upgrade.go:78  [T1,n1] 460  no need to upgrade, cluster already at the newest version
I230501 11:35:23.933499 909548 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 461  warmed privileges for virtual tables in 48.778893ms
I230501 11:35:23.999001 493708 3@pebble/event.go:717  [n1,s1,pebble] 462  [JOB 7] WAL created 000006
I230501 11:35:24.002823 485049 1@server/server_sql.go:1733  [T1,n1] 463  serving sql connections
--- FAIL: TestRefresh (6.96s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230501 11:35:24.099966 923296 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.101885 923615 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.103181 923647 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.104839 923647 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.107098 924720 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.108738 925289 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.110006 925661 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.111712 925970 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.113580 926388 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.115381 926786 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.117647 927113 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.119383 927505 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.122032 927782 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.123837 928223 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.125951 928607 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.127270 929032 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.128694 929345 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230501 11:35:24.130503 929636 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9907736?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9907736?buildTab=artifacts#/) on master @ [ed08739c22ecd246c9745d0f301976023c0d004b](https://github.com/cockroachdb/cockroach/commits/ed08739c22ecd246c9745d0f301976023c0d004b):


```
I230502 11:28:38.274261 885590 jobs/registry.go:1585  [T1,n1] 412  MIGRATION job 861655918181089281: stepping through state running
I230502 11:28:38.342368 498684 kv/kvserver/replica_rangefeed.go:709  [T1,n1,s1,r22/1:/Table/2{0-1}] 413  RangeFeed closed timestamp is empty
I230502 11:28:38.342823 891192 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861655918181089281,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 414 ={"Timestamp":1683026918326908635,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230502 11:28:38.362514 892597 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861655918181089281,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 415 ={"Timestamp":1683026918344353229,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230502 11:28:38.380577 895002 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861655918181089281,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 416 ={"Timestamp":1683026918364393739,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230502 11:28:38.399083 897116 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=861655918181089281,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 417 ={"Timestamp":1683026918382134333,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230502 11:28:38.407592 885590 jobs/registry.go:1585  [T1,n1] 418  MIGRATION job 861655918181089281: stepping through state succeeded
I230502 11:28:38.497216 489824 jobs/wait.go:145  [T1,n1] 419  waited for 1 [861655918181089281] queued jobs to complete 297.554779ms
I230502 11:28:38.498596 489824 upgrade/upgrademanager/manager.go:283  [T1,n1] 420  running permanent upgrade for version 1000022.2-102
I230502 11:28:38.669268 489824 upgrade/upgrademanager/manager.go:742  [T1,n1] 421  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230502 11:28:38.669436 914924 jobs/adopt.go:261  [T1,n1] 422  job 861655919816409089: resuming execution
I230502 11:28:38.743906 917561 jobs/registry.go:1585  [T1,n1] 423  MIGRATION job 861655919816409089: stepping through state running
I230502 11:28:38.848156 917561 jobs/registry.go:1585  [T1,n1] 424  MIGRATION job 861655919816409089: stepping through state succeeded
I230502 11:28:38.936666 489824 jobs/wait.go:145  [T1,n1] 425  waited for 1 [861655919816409089] queued jobs to complete 267.275ms
I230502 11:28:38.936760 489824 server/server_sql.go:1615  [T1,n1] 426  done ensuring all necessary startup migrations have run
I230502 11:28:38.938310 932037 jobs/job_scheduler.go:407  [T1,n1] 427  waiting 4m0s before scheduled jobs daemon start
I230502 11:28:38.942505 786401 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 428  tenant-settings-watcher: established range feed cache
I230502 11:28:38.946136 933633 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 429  tenant-capability-watcher: established range feed cache
I230502 11:28:38.955646 933633 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 430  received results of a full table scan for tenant capabilities
I230502 11:28:38.982094 932071 server/auto_upgrade.go:78  [T1,n1] 431  no need to upgrade, cluster already at the newest version
I230502 11:28:38.989200 931289 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 432  warmed privileges for virtual tables in 50.75567ms
I230502 11:28:39.034073 498492 3@pebble/event.go:717  [n1,s1,pebble] 433  [JOB 7] WAL created 000006
I230502 11:28:39.037499 489824 1@server/server_sql.go:1733  [T1,n1] 434  serving sql connections
--- FAIL: TestRefresh (6.50s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230502 11:28:39.147034 946794 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.149173 947342 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.151211 947265 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.152713 947265 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.154261 948450 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.155479 948686 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.156887 949024 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.158145 949290 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.159654 949630 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.160842 949756 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.162219 950404 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.163632 950752 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.164965 951082 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.166186 951420 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.167187 951740 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.168366 952115 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:28:39.169701 952482 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9907741?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9907741?buildTab=artifacts#/) on master @ [ed08739c22ecd246c9745d0f301976023c0d004b](https://github.com/cockroachdb/cockroach/commits/ed08739c22ecd246c9745d0f301976023c0d004b):


```
I230502 11:37:21.550691 5706 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.551885 5707 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.553042 5708 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.554229 5709 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.555392 5710 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.556663 5711 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.557900 5712 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.559048 5713 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.560196 5746 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.561388 5747 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.562538 5748 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.563696 5749 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.564839 5750 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.566005 5751 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.567130 5752 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.568347 5753 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.569499 5754 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.570641 5755 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.571800 5756 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.572975 5757 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.574206 5758 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.575303 5759 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.576426 5760 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.577610 5761 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.578793 5762 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.579911 5763 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.581082 5764 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.582247 5765 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.583443 5766 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.584596 5767 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.585725 5768 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.586874 5769 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.593571 5249 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.594776 5496 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.595902 5497 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.597001 5498 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.598197 5499 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:21.599299 5500 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230502 11:37:22.216883 3310 3@pebble/event.go:717  [n1,s1,pebble] 497  [JOB 7] WAL created 000006
I230502 11:37:22.240368 2920 1@server/server_sql.go:1733  [T1,n1] 498  serving sql connections
--- FAIL: TestRefresh (7.58s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9919654?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9919654?buildTab=artifacts#/) on master @ [9e0b06bf61700d5192f171c6ebaec7e8a10fa6e6](https://github.com/cockroachdb/cockroach/commits/9e0b06bf61700d5192f171c6ebaec7e8a10fa6e6):


```
I230503 11:10:16.736978 487982 server/server_sql.go:1615  [T1,n1] 394  done ensuring all necessary startup migrations have run
I230503 11:10:16.739063 911786 jobs/job_scheduler.go:407  [T1,n1] 395  waiting 3m0s before scheduled jobs daemon start
I230503 11:10:16.743389 912415 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 396  tenant-settings-watcher: established range feed cache
I230503 11:10:16.747248 913220 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 397  tenant-capability-watcher: established range feed cache
I230503 11:10:16.751822 913220 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 398  received results of a full table scan for tenant capabilities
I230503 11:10:16.776360 911861 server/auto_upgrade.go:78  [T1,n1] 399  no need to upgrade, cluster already at the newest version
I230503 11:10:16.785564 911796 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 400  warmed privileges for virtual tables in 46.40698ms
I230503 11:10:16.857073 497252 3@pebble/event.go:717  [n1,s1,pebble] 401  [JOB 7] WAL created 000006
I230503 11:10:16.861588 487982 1@server/server_sql.go:1733  [T1,n1] 402  serving sql connections
--- FAIL: TestRefresh (6.77s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230503 11:10:16.957024 925745 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 403  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.959035 926294 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 404  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.960503 926247 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 405  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.962841 926247 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 406  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.964931 927408 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 407  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.967286 927976 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 408  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.969594 928366 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 409  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.971052 928874 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 410  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.972908 929179 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 411  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.974845 929559 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 412  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.976339 929975 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 413  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.978009 930319 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 414  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.979811 930699 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 415  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.981792 931159 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 416  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.983458 931477 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 417  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.984992 931891 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 418  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.986770 932209 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 419  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.988161 932620 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 420  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.990204 932964 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 421  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.991781 933394 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 422  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.993114 933742 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 423  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.994207 933977 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 424  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.995552 934153 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 425  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.996307 934399 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 426  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.997023 934541 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 427  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.997979 934656 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 428  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.999156 934865 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 429  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:16.999973 935019 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 430  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:17.001096 935173 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 431  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:10:17.002559 934799 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 432  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
I230503 11:10:17.003751 935501 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 433  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9919659?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9919659?buildTab=artifacts#/) on master @ [9e0b06bf61700d5192f171c6ebaec7e8a10fa6e6](https://github.com/cockroachdb/cockroach/commits/9e0b06bf61700d5192f171c6ebaec7e8a10fa6e6):


```
I230503 11:16:25.578164 6067 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.579240 6068 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.580452 6069 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.581551 6070 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.582631 6071 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.583766 6072 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.584975 6073 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.586098 6074 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.587249 6075 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.588351 6076 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.589464 6077 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.590520 6078 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.591598 6079 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.592643 6080 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.593785 6081 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.594876 6082 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.596071 6083 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.597209 6084 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.598530 5784 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.599639 5785 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.600865 5786 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.601937 5787 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.603008 5788 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.604239 5789 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.605331 5790 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.606444 5791 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.607514 5792 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.608729 5793 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.610012 6098 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.611193 6099 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.612468 6100 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.613612 6101 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.614706 6102 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.615782 6103 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.616926 6104 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.618134 6105 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.619386 6106 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:25.620649 5596 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230503 11:16:26.278557 3512 3@pebble/event.go:717  [n1,s1,pebble] 498  [JOB 7] WAL created 000006
I230503 11:16:26.302674 2973 1@server/server_sql.go:1733  [T1,n1] 499  serving sql connections
--- FAIL: TestRefresh (7.27s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9932082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9932082?buildTab=artifacts#/) on master @ [5c9746012cb9a3244f47098254e603b7d924fdb2](https://github.com/cockroachdb/cockroach/commits/5c9746012cb9a3244f47098254e603b7d924fdb2):


```
I230504 11:59:06.138264 957546 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 432  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.139687 957696 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 433  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.144737 957907 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 434  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.146605 958406 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.165505 958638 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.168735 958991 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.170233 959378 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.171222 959648 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.173236 959925 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.175958 960311 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.177134 960679 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.178968 960965 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.179999 961273 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.182154 961404 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.183264 961796 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.184841 962148 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.185840 962277 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.186643 962443 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.187500 962416 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.188768 962766 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.190018 963012 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 452  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.190840 963207 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 453  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.191686 963036 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 454  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.193906 963520 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 455  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.197573 963832 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 456  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.199514 964045 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.200399 964187 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.201054 964285 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.201668 964353 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.202610 964421 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.203523 964503 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.204484 964619 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.205277 964703 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.205993 963647 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.206817 964795 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.207493 964947 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.208099 965015 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.208765 964943 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.209496 965187 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.210174 965287 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 11:59:06.210848 965403 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
I230504 11:59:06.211856 964059 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.20s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9932089?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9932089?buildTab=artifacts#/) on master @ [5c9746012cb9a3244f47098254e603b7d924fdb2](https://github.com/cockroachdb/cockroach/commits/5c9746012cb9a3244f47098254e603b7d924fdb2):


```
I230504 12:05:23.896665 5150 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.897631 5151 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.898777 5398 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.899896 5399 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.901035 5400 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.902178 5401 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.903180 5402 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.904235 5403 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.905272 5404 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.906367 5405 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.907570 5406 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.908573 5407 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.909669 5408 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.910838 5409 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.911844 5410 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.912811 5152 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.915109 5385 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.916562 5051 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.917952 5052 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.919243 5053 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.921910 5358 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.923346 5166 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.924590 5167 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.925878 5168 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.927286 5242 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.928531 5243 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.929828 5244 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.931262 5359 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.932534 5360 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.933909 5361 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.935199 5426 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.936453 5427 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.937582 5259 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.938600 5260 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.939670 5261 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.940566 5262 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.941505 5263 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:23.942473 5264 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230504 12:05:24.406592 2988 3@pebble/event.go:717  [n1,s1,pebble] 497  [JOB 7] WAL created 000006
I230504 12:05:24.422570 2818 1@server/server_sql.go:1733  [T1,n1] 498  serving sql connections
--- FAIL: TestRefresh (6.15s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9944179?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9944179?buildTab=artifacts#/) on master @ [918923738a73c39986c163465edb70e9af2c49b8](https://github.com/cockroachdb/cockroach/commits/918923738a73c39986c163465edb70e9af2c49b8):


```
I230505 11:04:02.740013 945006 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 428  tenant-settings-watcher: established range feed cache
I230505 11:04:02.746169 945586 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 429  tenant-capability-watcher: established range feed cache
I230505 11:04:02.751408 945586 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 430  received results of a full table scan for tenant capabilities
I230505 11:04:02.779229 944274 server/auto_upgrade.go:78  [T1,n1] 431  no need to upgrade, cluster already at the newest version
I230505 11:04:02.788325 944221 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 432  warmed privileges for virtual tables in 55.774983ms
I230505 11:04:02.898026 509472 3@pebble/event.go:717  [n1,s1,pebble] 433  [JOB 7] WAL created 000006
I230505 11:04:02.902778 500220 1@server/server_sql.go:1733  [T1,n1] 434  serving sql connections
--- FAIL: TestRefresh (6.89s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230505 11:04:02.985086 958158 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:02.988029 958538 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:02.989985 958525 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:02.991225 958525 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:02.998179 959470 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:02.999180 959744 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.000069 959841 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.000811 959973 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.001483 960088 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.002303 960175 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.003004 960291 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.004656 960398 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.005363 960601 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.021404 961307 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.026350 962536 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.027845 963019 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.029523 963374 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.030673 963840 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 452  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.032646 964184 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 453  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.033774 964587 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 454  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.035147 964934 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 455  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.037004 965259 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 456  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.038334 965814 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.039329 966155 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.040507 966401 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.041568 966741 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.042665 966987 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.043743 967313 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.045076 967592 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.046535 967787 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.048599 967972 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.049577 968178 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:04:03.050994 968408 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9944187?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9944187?buildTab=artifacts#/) on master @ [918923738a73c39986c163465edb70e9af2c49b8](https://github.com/cockroachdb/cockroach/commits/918923738a73c39986c163465edb70e9af2c49b8):


```
I230505 11:18:09.167499 5429 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.168516 5430 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.169493 5431 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.170400 5432 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.171386 5433 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.172483 5443 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.173472 5444 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.174483 5445 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.175512 5446 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.176505 5447 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.177475 5421 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.178376 5422 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.179268 5423 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.180280 5424 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.181256 5434 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.182245 5435 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.183212 5436 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.184264 5448 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.185242 5449 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.186191 5425 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.187104 5458 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.188077 5459 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.189087 5437 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.190346 5450 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.191489 5460 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.192424 5461 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.193455 5462 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.194451 5438 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.195704 5451 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.196865 5463 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.198110 5439 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.199456 5452 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.200627 5464 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.201652 5440 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.202621 5441 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.203622 5474 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.204904 5453 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.206103 5465 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230505 11:18:09.514667 2987 3@pebble/event.go:717  [n1,s1,pebble] 498  [JOB 7] WAL created 000006
I230505 11:18:09.537485 2818 1@server/server_sql.go:1733  [T1,n1] 499  serving sql connections
--- FAIL: TestRefresh (6.49s)
=== RUN   TestRefresh/needs_refresh,_no_change
    cache_test.go:185: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:627
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:185
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestRefresh/needs_refresh,_no_change
    --- FAIL: TestRefresh/needs_refresh,_no_change (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9966219?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9966219?buildTab=artifacts#/) on master @ [53abb2f174283b3de637774e3dbbaeeaa23cde65](https://github.com/cockroachdb/cockroach/commits/53abb2f174283b3de637774e3dbbaeeaa23cde65):


```
I230507 10:16:59.317369 5413 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.318338 5414 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.319368 5303 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.320332 5304 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.321235 5305 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.322174 5306 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.323042 5307 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.323978 5308 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.324950 5309 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.325911 5161 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.326859 5162 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.327710 5163 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.328651 5164 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.329567 5165 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.330475 5166 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.331406 5167 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.332368 5168 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.333290 5169 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.334163 5426 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.335084 5427 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.336086 5310 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.337019 5311 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.337927 5312 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.338864 5313 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.339910 5369 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.340899 5370 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.341830 5428 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.342755 5429 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.343718 5430 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.344769 5431 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.345779 5432 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.346805 5433 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.347918 5434 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.348930 5435 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.350046 5436 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.351078 5437 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.352228 5442 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.353141 5443 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:16:59.801830 2944 3@pebble/event.go:717  [n1,s1,pebble] 498  [JOB 7] WAL created 000006
I230507 10:16:59.821008 2728 1@server/server_sql.go:1733  [T1,n1] 499  serving sql connections
--- FAIL: TestRefresh (5.90s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9966215?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9966215?buildTab=artifacts#/) on master @ [53abb2f174283b3de637774e3dbbaeeaa23cde65](https://github.com/cockroachdb/cockroach/commits/53abb2f174283b3de637774e3dbbaeeaa23cde65):


```
I230507 10:17:48.329255 947487 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 429  tenant-capability-watcher: established range feed cache
I230507 10:17:48.354173 947487 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 430  received results of a full table scan for tenant capabilities
I230507 10:17:48.391014 946770 server/auto_upgrade.go:78  [T1,n1] 431  no need to upgrade, cluster already at the newest version
I230507 10:17:48.401467 946753 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 432  warmed privileges for virtual tables in 82.872997ms
I230507 10:17:48.459165 519773 3@pebble/event.go:717  [n1,s1,pebble] 433  [JOB 7] WAL created 000006
I230507 10:17:48.465118 510707 1@server/server_sql.go:1733  [T1,n1] 434  serving sql connections
--- FAIL: TestRefresh (6.76s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230507 10:17:48.563546 962965 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.567562 963472 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.569196 963385 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.571018 963385 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.572233 964699 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.575958 964935 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.577082 966033 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.581395 966365 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.583547 966879 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.585696 967299 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.588358 967936 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.594143 968446 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.594872 969554 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.595573 969648 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.596281 969734 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.596829 969802 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.597416 969870 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.597977 969958 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 452  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.598656 970026 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 453  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.599362 970120 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 454  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.599938 970197 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 455  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.600482 970265 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 456  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.601002 970333 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.601538 970401 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.602131 970468 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.602853 970562 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.603519 970537 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.604344 970691 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.606303 970828 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.607673 971046 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.608889 971164 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.609679 971353 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.612656 971480 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230507 10:17:48.614342 971808 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977923?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977923?buildTab=artifacts#/) on master @ [adad57859a5281e2846082761c8232890b3d2a68](https://github.com/cockroachdb/cockroach/commits/adad57859a5281e2846082761c8232890b3d2a68):


```
I230508 11:03:07.590334 5637 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.591284 5638 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.592229 5639 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.593287 5640 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.594260 5641 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.595181 5642 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.596104 5643 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.597041 5644 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.597933 5645 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.598908 5646 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.599867 5647 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.600870 5648 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.601873 5649 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.603118 5577 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.604246 5650 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.605273 5578 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.606258 5579 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.607262 5596 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.608167 5597 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.609089 5598 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.609976 5599 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.610998 5600 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.611926 5601 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.612883 5666 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.613824 5667 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.614711 5668 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.615593 5669 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.616582 5670 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.617588 5671 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.618522 5672 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.619429 5673 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.620407 5674 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.621348 5675 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.622288 5676 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.623199 5677 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.624156 5678 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.625172 5679 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:07.626191 5680 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:03:08.100559 3096 3@pebble/event.go:717  [n1,s1,pebble] 497  [JOB 7] WAL created 000006
I230508 11:03:08.124209 2901 1@server/server_sql.go:1733  [T1,n1] 498  serving sql connections
--- FAIL: TestRefresh (6.11s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977918?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9977918?buildTab=artifacts#/) on master @ [adad57859a5281e2846082761c8232890b3d2a68](https://github.com/cockroachdb/cockroach/commits/adad57859a5281e2846082761c8232890b3d2a68):


```
I230508 11:04:03.893443 504707 upgrade/upgrademanager/manager.go:742  [T1,n1] 412  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230508 11:04:03.893633 898688 jobs/adopt.go:261  [T1,n1] 413  job 863349780412760065: resuming execution
I230508 11:04:03.908799 514856 kv/kvserver/replica_rangefeed.go:709  [T1,n1,s1,r22/1:/Table/2{0-1}] 414  RangeFeed closed timestamp is empty
I230508 11:04:03.964171 901007 jobs/registry.go:1585  [T1,n1] 415  MIGRATION job 863349780412760065: stepping through state running
I230508 11:04:04.032987 906686 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=863349780412760065,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 416 ={"Timestamp":1683543844014323861,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230508 11:04:04.051068 907966 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=863349780412760065,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 417 ={"Timestamp":1683543844034752240,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230508 11:04:04.069823 910194 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=863349780412760065,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 418 ={"Timestamp":1683543844053360768,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230508 11:04:04.099183 912345 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=863349780412760065,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 419 ={"Timestamp":1683543844073792658,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230508 11:04:04.108054 901007 jobs/registry.go:1585  [T1,n1] 420  MIGRATION job 863349780412760065: stepping through state succeeded
I230508 11:04:04.192694 504707 jobs/wait.go:145  [T1,n1] 421  waited for 1 [863349780412760065] queued jobs to complete 299.095117ms
I230508 11:04:04.194024 504707 upgrade/upgrademanager/manager.go:283  [T1,n1] 422  running permanent upgrade for version 1000022.2-102
I230508 11:04:04.352417 504707 upgrade/upgrademanager/manager.go:742  [T1,n1] 423  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230508 11:04:04.352679 930449 jobs/adopt.go:261  [T1,n1] 424  job 863349781935259649: resuming execution
I230508 11:04:04.422245 932854 jobs/registry.go:1585  [T1,n1] 425  MIGRATION job 863349781935259649: stepping through state running
I230508 11:04:04.569607 932854 jobs/registry.go:1585  [T1,n1] 426  MIGRATION job 863349781935259649: stepping through state succeeded
I230508 11:04:04.660972 504707 jobs/wait.go:145  [T1,n1] 427  waited for 1 [863349781935259649] queued jobs to complete 308.34672ms
I230508 11:04:04.661088 504707 server/server_sql.go:1615  [T1,n1] 428  done ensuring all necessary startup migrations have run
I230508 11:04:04.665044 947359 jobs/job_scheduler.go:407  [T1,n1] 429  waiting 3m0s before scheduled jobs daemon start
I230508 11:04:04.666878 947745 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 430  tenant-settings-watcher: established range feed cache
I230508 11:04:04.675641 948869 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 431  tenant-capability-watcher: established range feed cache
I230508 11:04:04.679108 948869 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 432  received results of a full table scan for tenant capabilities
I230508 11:04:04.700978 947386 server/auto_upgrade.go:78  [T1,n1] 433  no need to upgrade, cluster already at the newest version
I230508 11:04:04.721078 947369 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 434  warmed privileges for virtual tables in 56.246172ms
I230508 11:04:04.792383 514346 3@pebble/event.go:717  [n1,s1,pebble] 435  [JOB 7] WAL created 000006
I230508 11:04:04.796084 504707 1@server/server_sql.go:1733  [T1,n1] 436  serving sql connections
--- FAIL: TestRefresh (6.87s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230508 11:04:04.917460 962126 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.920254 962679 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.921361 962597 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.922226 962597 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.923647 963726 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.924956 964055 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.926579 964474 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.928236 964819 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.930181 965290 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.936992 965671 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.938165 967340 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.940067 967837 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.941777 968325 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.943006 968666 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230508 11:04:04.944480 969080 13@kv/kvserver/replicate_queue.go:839  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 4
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9993723?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9993723?buildTab=artifacts#/) on master @ [439c515b2a0058648731da73993c409544404da1](https://github.com/cockroachdb/cockroach/commits/439c515b2a0058648731da73993c409544404da1):


```
I230509 10:33:00.429011 975704 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 429  tenant-capability-watcher: established range feed cache
I230509 10:33:00.431601 975704 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 430  received results of a full table scan for tenant capabilities
I230509 10:33:00.448722 974327 server/auto_upgrade.go:78  [T1,n1] 431  no need to upgrade, cluster already at the newest version
I230509 10:33:00.460120 974214 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 432  warmed privileges for virtual tables in 39.741727ms
I230509 10:33:00.575838 525248 3@pebble/event.go:717  [n1,s1,pebble] 433  [JOB 7] WAL created 000006
I230509 10:33:00.579604 516140 1@server/server_sql.go:1733  [T1,n1] 434  serving sql connections
--- FAIL: TestRefresh (7.00s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230509 10:33:00.656593 987624 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.657985 989054 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.659412 989423 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.661614 989047 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.663681 989047 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.665086 989047 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.667020 990954 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.668925 991493 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.670858 991908 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.672240 992316 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.674510 992727 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.677568 993345 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.679738 994121 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.681364 994755 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.683196 995197 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.684495 995396 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.685742 995597 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.686938 995722 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 452  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.688671 995862 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 453  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.689388 995988 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 454  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.690046 996087 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 455  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.690730 996188 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 456  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.691358 996321 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.691946 996434 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.692698 996518 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.693269 996618 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.693931 996718 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.694524 996802 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.695216 996902 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.695827 997002 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.696507 997143 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.697106 997249 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.697739 997349 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:00.698361 997432 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9993726?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9993726?buildTab=artifacts#/) on master @ [439c515b2a0058648731da73993c409544404da1](https://github.com/cockroachdb/cockroach/commits/439c515b2a0058648731da73993c409544404da1):


```
I230509 10:33:55.165796 5450 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.166968 5451 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.168017 5452 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.169036 5453 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.170034 5454 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.171236 5455 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.172367 5456 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.173509 5422 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.174584 5423 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.175593 5424 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.176716 5425 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.177693 5458 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.178835 5459 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.179954 5460 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.181091 5461 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.182177 5462 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.183076 5463 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.184004 5464 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.185008 5465 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.185986 5466 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.187119 5467 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.188051 5468 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.189049 5469 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.189907 5470 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.190934 5471 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.191985 5431 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.193066 5432 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.194130 5472 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.195212 5473 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.196257 5474 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.197314 5475 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.198278 5476 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.199463 5477 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.200536 5478 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.201497 5479 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.202538 5480 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.203706 5481 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.204721 5482 13@kv/kvserver/replicate_queue.go:756  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230509 10:33:55.650080 2908 3@pebble/event.go:717  [n1,s1,pebble] 496  [JOB 7] WAL created 000006
I230509 10:33:55.666497 2806 1@server/server_sql.go:1733  [T1,n1] 497  serving sql connections
--- FAIL: TestRefresh (6.30s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10064609?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10064609?buildTab=artifacts#/) on master @ [63eb6b86c1518fbed9db10e7d2afa1446b2b3b82](https://github.com/cockroachdb/cockroach/commits/63eb6b86c1518fbed9db10e7d2afa1446b2b3b82):


```
I230513 10:54:18.605180 856700 jobs/registry.go:1592  [T1,n1] 403  MIGRATION job 864763439472607233: stepping through state succeeded
I230513 10:54:18.684555 871788 jobs/registry.go:1592  [T1,n1] 404  AUTO CONFIG RUNNER job 102: stepping through state running
I230513 10:54:18.703111 490294 jobs/wait.go:145  [T1,n1] 405  waited for 1 [864763439472607233] queued jobs to complete 295.356641ms
I230513 10:54:18.704624 490294 upgrade/upgrademanager/manager.go:283  [T1,n1] 406  running permanent upgrade for version 1000022.2-98
I230513 10:54:18.740365 871788 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 407  using provider with type acprovider.NoTaskProvider
I230513 10:54:18.740445 871788 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 408  waiting for environment activation...
I230513 10:54:18.740534 871788 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 409  waiting for environment activation...
I230513 10:54:18.906621 490294 upgrade/upgrademanager/manager.go:742  [T1,n1] 410  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230513 10:54:18.906914 890978 jobs/adopt.go:261  [T1,n1] 411  job 864763441003560961: resuming execution
I230513 10:54:18.907651 498735 kv/kvserver/replica_rangefeed.go:709  [T1,n1,s1,r34/1:/Table/3{2-3}] 412  RangeFeed closed timestamp is empty
I230513 10:54:18.977527 893231 jobs/registry.go:1592  [T1,n1] 413  MIGRATION job 864763441003560961: stepping through state running
I230513 10:54:19.044571 898794 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=864763441003560961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 414 ={"Timestamp":1683975259028421371,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230513 10:54:19.068290 900171 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=864763441003560961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 415 ={"Timestamp":1683975259051209527,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230513 10:54:19.092455 902616 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=864763441003560961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 416 ={"Timestamp":1683975259079041138,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230513 10:54:19.112164 904527 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=864763441003560961,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 417 ={"Timestamp":1683975259094973059,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230513 10:54:19.129486 893231 jobs/registry.go:1592  [T1,n1] 418  MIGRATION job 864763441003560961: stepping through state succeeded
I230513 10:54:19.219066 490294 jobs/wait.go:145  [T1,n1] 419  waited for 1 [864763441003560961] queued jobs to complete 312.190722ms
I230513 10:54:19.220366 490294 upgrade/upgrademanager/manager.go:283  [T1,n1] 420  running permanent upgrade for version 1000022.2-102
I230513 10:54:19.426867 490294 upgrade/upgrademanager/manager.go:742  [T1,n1] 421  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230513 10:54:19.427154 922533 jobs/adopt.go:261  [T1,n1] 422  job 864763442693537793: resuming execution
I230513 10:54:19.506535 924959 jobs/registry.go:1592  [T1,n1] 423  MIGRATION job 864763442693537793: stepping through state running
I230513 10:54:19.605386 924959 jobs/registry.go:1592  [T1,n1] 424  MIGRATION job 864763442693537793: stepping through state succeeded
I230513 10:54:19.689878 490294 jobs/wait.go:145  [T1,n1] 425  waited for 1 [864763442693537793] queued jobs to complete 262.832298ms
I230513 10:54:19.689991 490294 server/server_sql.go:1616  [T1,n1] 426  done ensuring all necessary startup migrations have run
I230513 10:54:19.693068 940412 jobs/job_scheduler.go:407  [T1,n1] 427  waiting 4m0s before scheduled jobs daemon start
I230513 10:54:19.700000 941451 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 428  tenant-settings-watcher: established range feed cache
I230513 10:54:19.705945 942165 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 429  tenant-capability-watcher: established range feed cache
I230513 10:54:19.708973 942165 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 430  received results of a full table scan for tenant capabilities
I230513 10:54:19.728067 940508 server/auto_upgrade.go:78  [T1,n1] 431  no need to upgrade, cluster already at the newest version
I230513 10:54:19.746913 940438 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 432  warmed privileges for virtual tables in 53.695875ms
I230513 10:54:19.804417 498302 3@pebble/event.go:717  [n1,s1,pebble] 433  [JOB 7] WAL created 000006
I230513 10:54:19.808581 490294 1@server/server_sql.go:1734  [T1,n1] 434  serving sql connections
--- FAIL: TestRefresh (6.80s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230513 10:54:19.926504 955254 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.933477 955691 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.934649 955644 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.935887 955644 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.938927 958136 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.940374 958474 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.941638 958872 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:54:19.943670 959225 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10064614?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10064614?buildTab=artifacts#/) on master @ [63eb6b86c1518fbed9db10e7d2afa1446b2b3b82](https://github.com/cockroachdb/cockroach/commits/63eb6b86c1518fbed9db10e7d2afa1446b2b3b82):


```
I230513 10:56:41.339392 5411 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 457  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.340441 5412 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 458  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.341450 5413 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 459  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.342379 5414 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.343452 5415 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.344409 5416 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.345436 5417 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.346332 5418 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.347421 5419 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.348483 5420 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.349530 5421 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.350591 5422 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.351527 5423 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.352735 5424 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.353757 5425 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.354935 5426 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.355906 5399 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.356982 5344 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.358110 5345 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.359107 5427 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.360217 5428 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.361401 5442 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.362477 5443 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.363568 5429 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.364488 5430 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.365419 5431 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.366486 5432 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.367503 5433 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.368503 5434 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.369473 5435 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.370488 5436 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.371451 5437 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.372546 5438 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.373637 5439 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.374622 5440 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.375612 5441 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.376586 5458 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:41.377725 5459 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230513 10:56:42.023756 2961 3@pebble/event.go:717  [n1,s1,pebble] 495  [JOB 7] WAL created 000006
I230513 10:56:42.040193 2763 1@server/server_sql.go:1734  [T1,n1] 496  serving sql connections
--- FAIL: TestRefresh (6.42s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10075211?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10075211?buildTab=artifacts#/) on master @ [63eb6b86c1518fbed9db10e7d2afa1446b2b3b82](https://github.com/cockroachdb/cockroach/commits/63eb6b86c1518fbed9db10e7d2afa1446b2b3b82):


```
I230514 11:12:39.164025 495701 jobs/wait.go:145  [T1,n1] 417  waited for 1 [865050160908697601] queued jobs to complete 301.222799ms
I230514 11:12:39.165345 495701 upgrade/upgrademanager/manager.go:283  [T1,n1] 418  running permanent upgrade for version 1000022.2-102
I230514 11:12:39.341017 495701 upgrade/upgrademanager/manager.go:742  [T1,n1] 419  running Upgrade to 1000022.2-102: "create sql activity updater job"
I230514 11:12:39.341422 920836 jobs/adopt.go:261  [T1,n1] 420  job 865050162513248257: resuming execution
I230514 11:12:39.412750 923038 jobs/registry.go:1592  [T1,n1] 421  MIGRATION job 865050162513248257: stepping through state running
I230514 11:12:39.521328 923038 jobs/registry.go:1592  [T1,n1] 422  MIGRATION job 865050162513248257: stepping through state succeeded
I230514 11:12:39.607307 495701 jobs/wait.go:145  [T1,n1] 423  waited for 1 [865050162513248257] queued jobs to complete 266.174627ms
I230514 11:12:39.607418 495701 server/server_sql.go:1616  [T1,n1] 424  done ensuring all necessary startup migrations have run
I230514 11:12:39.609915 938989 jobs/job_scheduler.go:407  [T1,n1] 425  waiting 2m0s before scheduled jobs daemon start
I230514 11:12:39.612163 939341 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 426  tenant-settings-watcher: established range feed cache
I230514 11:12:39.616055 940019 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 427  tenant-capability-watcher: established range feed cache
I230514 11:12:39.619469 940019 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 428  received results of a full table scan for tenant capabilities
I230514 11:12:39.658697 938999 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 429  warmed privileges for virtual tables in 48.378599ms
I230514 11:12:39.663279 939032 server/auto_upgrade.go:78  [T1,n1] 430  no need to upgrade, cluster already at the newest version
I230514 11:12:39.753805 505339 3@pebble/event.go:717  [n1,s1,pebble] 431  [JOB 7] WAL created 000006
I230514 11:12:39.760551 495701 1@server/server_sql.go:1734  [T1,n1] 432  serving sql connections
--- FAIL: TestRefresh (6.99s)
=== RUN   TestRefresh/needs_refresh,_with_change
I230514 11:12:39.884624 952760 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 433  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.886570 953028 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 434  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.888371 952996 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 435  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.889690 952996 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 436  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.891676 954184 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 437  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.893307 954452 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r5/1:/{Systemtse-Table/0}] 438  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.895057 954754 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 439  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.896442 955043 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 440  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.898230 955434 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r8/1:/Table/{4-5}] 441  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.899623 955900 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r9/1:/Table/{5-6}] 442  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.901876 956178 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 443  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.903015 956620 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 444  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.904009 956881 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 445  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.905061 957133 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 446  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.906375 957525 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r15/1:/Table/1{3-4}] 447  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.907505 957861 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 448  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.911925 958153 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r16/1:/Table/1{4-5}] 449  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.913348 959293 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r17/1:/Table/1{5-6}] 450  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.915352 959825 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r18/1:/Table/1{6-7}] 451  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.916661 960206 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r19/1:/Table/1{7-8}] 452  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.918994 960585 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 453  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.920291 961166 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r21/1:/Table/{19-20}] 454  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.921001 961312 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 455  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:12:39.922708 961433 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 456  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/kv/kvserver/protectedts/ptcache/ptcache_test.TestRefresh [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10075216?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10075216?buildTab=artifacts#/) on master @ [63eb6b86c1518fbed9db10e7d2afa1446b2b3b82](https://github.com/cockroachdb/cockroach/commits/63eb6b86c1518fbed9db10e7d2afa1446b2b3b82):


```
I230514 11:15:00.761119 5426 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r26/1:/Table/2{4-5}] 460  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.762088 5427 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r27/1:/Table/2{5-6}] 461  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.763058 5428 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r28/1:/Table/2{6-7}] 462  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.764102 5247 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r29/1:/Table/2{7-8}] 463  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.765168 5248 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r30/1:/Table/2{8-9}] 464  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.766233 5222 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 465  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.767320 5249 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r33/1:/{NamespaceTab…-Table/32}] 466  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.768351 5442 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r32/1:/NamespaceTable/{30-Max}] 467  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.769316 5443 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r34/1:/Table/3{2-3}] 468  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.770271 5118 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 469  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.771146 5119 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r36/1:/Table/3{4-5}] 470  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.772102 5120 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r37/1:/Table/3{5-6}] 471  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.773059 5121 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r38/1:/Table/3{6-7}] 472  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.773967 5458 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 473  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.775059 5459 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r40/1:/Table/3{8-9}] 474  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.775992 5460 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 475  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.776945 5461 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 476  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.777882 5462 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 477  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.778803 5463 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 478  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.779724 5464 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 479  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.780682 5465 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r46/1:/Table/4{4-5}] 480  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.781567 5466 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r47/1:/Table/4{5-6}] 481  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.782525 5467 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r48/1:/Table/4{6-7}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.783435 5468 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r50/1:/Table/{48-50}] 483  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.784374 5469 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 484  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.785326 5470 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 485  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.786381 5471 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 486  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.787312 5472 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r53/1:/Table/5{2-3}] 487  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.788271 5473 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r54/1:/Table/5{3-4}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.789189 5474 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r55/1:/Table/5{4-5}] 489  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.790037 5475 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.790951 5476 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.791877 5477 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 492  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.792863 5478 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.793850 5479 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r60/1:/Table/{59-60}] 494  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.794757 5480 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r63/1:/{Table/62-Max}] 495  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.795682 5481 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r62/1:/Table/6{1-2}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:00.796586 5482 13@kv/kvserver/replicate_queue.go:764  [T1,n1,replicate,s1,r61/1:/Table/6{0-1}] 497  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230514 11:15:01.230367 2968 3@pebble/event.go:717  [n1,s1,pebble] 498  [JOB 7] WAL created 000006
I230514 11:15:01.256363 2779 1@server/server_sql.go:1734  [T1,n1] 499  serving sql connections
--- FAIL: TestRefresh (6.38s)
=== RUN   TestRefresh/needs_refresh,_with_change
    cache_test.go:191: 
        	Error Trace:	pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:626
        	            				pkg/kv/kvserver/protectedts/ptcache/ptcache_test_test/pkg/kv/kvserver/protectedts/ptcache/cache_test.go:191
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRefresh/needs_refresh,_with_change
    --- FAIL: TestRefresh/needs_refresh,_with_change (0.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101298 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1.0]
- #100418 pkg/kv/kvserver/protectedts/ptcache/ptcache_test: TestRefresh failed [C-test-failure O-robot branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRefresh.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/protectedts/ptcache/cache_test.go
+++ b/pkg/kv/kvserver/protectedts/ptcache/cache_test.go
@@ -152,6 +152,9 @@ func TestRefresh(t *testing.T) {
 	}
 	s, _, _ := serverutils.StartServer(t,
 		base.TestServerArgs{
+			// Disable span configs to avoid measuring protected timestamp lookups
+			// performed by the AUTO SPAN CONFIG RECONCILIATION job.
+			DisableSpanConfigs: true,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
 					TestingRequestFilter: st.requestFilter,

